rest/security-rest-cxfwrapper/src/main/java/org/codice/ddf/cxf/SecureCxfClientFactory.java

```diff
@@ -0,0 +1,275 @@
+import org.apache.cxf.interceptor.LoggingOutInterceptor;
+import org.apache.cxf.jaxrs.client.Client;
+import org.apache.cxf.jaxrs.client.ClientConfiguration;
+import org.apache.cxf.jaxrs.client.JAXRSClientFactoryBean;
+import org.apache.cxf.jaxrs.client.WebClient;
+import org.apache.cxf.message.Message;
+import org.apache.cxf.transport.http.HTTPConduit;
+import org.apache.shiro.subject.Subject;
+import org.apache.ws.security.WSSecurityException;
+import org.apache.ws.security.components.crypto.CredentialException;
+import org.apache.ws.security.components.crypto.Merlin;
+import org.codice.ddf.security.common.PropertiesLoader;
+import org.codice.ddf.security.handler.api.PKIAuthenticationToken;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Arrays;
+import java.util.List;
+
+public class SecureCxfClientFactory<T> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SecureCxfClientFactory.class);
+
+    private final Client cxfClient;
+
+    /**
+     * @see #SecureCxfClientFactory(String, Class, String, String, java.util.List, boolean)
+     */
+    public SecureCxfClientFactory(String endpointUrl, Class<T> interfaceClass)
+            throws SecurityServiceException {
+        this(endpointUrl, interfaceClass, null, null);
+    }
+
+    /**
+     * @see #SecureCxfClientFactory(String, Class, String, String, java.util.List, boolean)
+     */
+    public SecureCxfClientFactory(String endpointUrl, Class<T> interfaceClass, String username,
+            String password) throws SecurityServiceException {
+        this(endpointUrl, interfaceClass, username, password, null, false);
+    }
+
+    /**
+     * Constructs a factory that will return security-aware cxf clients. Once constructed,
+     * use the getClient* methods to retrieve a fresh client  with the same configuration.
+     * <p/>
+     * This factory can and should be cached. The clients it constructs should not be.
+     *
+     * @param endpointUrl    the remote url to connect to
+     * @param interfaceClass an interface representing the resource at the remote url
+     * @param username       optional username for basic auth
+     * @param password       optional password for basic auth
+     * @param providers      optional list of providers to further configure the client
+     * @param disableCnCheck disable ssl check for common name / host name match
+     */
+    public SecureCxfClientFactory(String endpointUrl, Class<T> interfaceClass, String username,
```

Comment:
[1mdiff --git opensearch/catalog-opensearch-source/pom.xml opensearch/catalog-opensearch-source/pom.xml[m
[1mindex 8ebf797..2ad2ee1 100644[m
[1m--- opensearch/catalog-opensearch-source/pom.xml[m
[1m+++ opensearch/catalog-opensearch-source/pom.xml[m
[36m@@ -159,6 +159,12 @@[m
             <version>${ddf.security.app.version}</version>[m
         </dependency>[m
 [m
[32m+[m[32m        <dependency>[m
[32m+[m[32m            <groupId>ddf.security</groupId>[m
[32m+[m[32m            <artifactId>security-rest-cxfwrapper</artifactId>[m
[32m+[m[32m            <version>${ddf.security.app.version}</version>[m
[32m+[m[32m        </dependency>[m
[32m+[m
     </dependencies>[m
 [m
     <build>[m
[36m@@ -183,8 +189,13 @@[m
                             org.apache.abdera.i18n.iri, <!-- Element -->[m
                             org.apache.abdera.i18n.rfc4646, <!-- Element -->[m
                             com.ctc.wstx.stax, <!-- needed by FOMWriter -->[m
[32m+[m[32m                            org.codice.ddf.security.handler.api;resolution:=optional,[m
[32m+[m[32m                            org.codice.ddf.security.common;resolution:=optional,[m
                             *[m
                         </Import-Package>[m
[32m+[m[32m                        <Embed-Dependency>[m
[32m+[m[32m                            security-rest-cxfwrapper[m
[32m+[m[32m                        </Embed-Dependency>[m
                         <Private-Package>[m
                             ddf.catalog.source.opensearch,[m
                             ddf.catalog.data.impl,[m
[1mdiff --git opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchConnection.java opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchConnection.java[m
[1mindex 388b730..7fc7eda 100644[m
[1m--- opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchConnection.java[m
[1m+++ opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchConnection.java[m
[36m@@ -17,14 +17,12 @@[m
 import ddf.catalog.filter.FilterAdapter;[m
 import ddf.catalog.operation.Query;[m
 import ddf.catalog.source.UnsupportedQueryException;[m
[31m-import ddf.security.settings.SecuritySettingsService;[m
[32m+[m[32mimport ddf.security.Subject;[m
[32m+[m[32mimport ddf.security.service.SecurityServiceException;[m
 import org.apache.commons.lang.StringUtils;[m
[31m-import org.apache.cxf.configuration.jsse.TLSClientParameters;[m
 import org.apache.cxf.jaxrs.client.Client;[m
[31m-import org.apache.cxf.jaxrs.client.ClientConfiguration;[m
[31m-import org.apache.cxf.jaxrs.client.JAXRSClientFactory;[m
 import org.apache.cxf.jaxrs.client.WebClient;[m
[31m-import org.apache.cxf.transport.http.HTTPConduit;[m
[32m+[m[32mimport org.codice.ddf.cxf.SecureCxfClientFactory;[m
 import org.codice.ddf.endpoints.OpenSearch;[m
 import org.codice.ddf.endpoints.rest.RESTService;[m
 import org.slf4j.Logger;[m
[36m@@ -43,13 +41,9 @@[m
     private static final transient Logger LOGGER = LoggerFactory[m
             .getLogger(OpenSearchConnection.class);[m
 [m
[31m-    protected OpenSearch openSearch;[m
[32m+[m[32m    protected SecureCxfClientFactory<OpenSearch> openSearchClient;[m
 [m
[31m-    protected RESTService restService;[m
[31m-[m
[31m-    protected Client openSearchClient;[m
[31m-[m
[31m-    protected Client restServiceClient;[m
[32m+[m[32m    protected SecureCxfClientFactory restServiceClient;[m
 [m
     private FilterAdapter filterAdapter;[m
 [m
[36m@@ -57,39 +51,32 @@[m
 [m
     private String password;[m
 [m
[31m-    private SecuritySettingsService securitySettingsService;[m
[31m-[m
     /**[m
      * Default Constructor[m
[31m-     * @param endpointUrl - OpenSearch URL to connect to[m
[32m+[m[32m     *[m
[32m+[m[32m     * @param endpointUrl   - OpenSearch URL to connect to[m
      * @param filterAdapter - adapter to translate between DDF REST and OpenSearch[m
[31m-     * @param securitySettings - Service used to obtain settings for secure communications.[m
[31m-     * @param username - Basic Auth user name[m
[31m-     * @param password - Basic Auth password[m
[32m+[m[32m     * @param username      - Basic Auth user name[m
[32m+[m[32m     * @param password      - Basic Auth password[m
      */[m
[31m-    public OpenSearchConnection(String endpointUrl, FilterAdapter filterAdapter,[m
[31m-            SecuritySettingsService securitySettings, String username, String password) {[m
[32m+[m[32m    public OpenSearchConnection(String endpointUrl, FilterAdapter filterAdapter, String username,[m
[32m+[m[32m            String password) throws SecurityServiceException {[m
         this.filterAdapter = filterAdapter;[m
         this.username = username;[m
         this.password = password;[m
[31m-        this.securitySettingsService = securitySettings;[m
[31m-        openSearch = JAXRSClientFactory.create(endpointUrl, OpenSearch.class);[m
[31m-        openSearchClient = WebClient.client(openSearch);[m
[32m+[m[32m        openSearchClient = new SecureCxfClientFactory<OpenSearch>(endpointUrl, OpenSearch.class,[m
[32m+[m[32m                username, password);[m
 [m
         RestUrl restUrl = newRestUrl(endpointUrl);[m
         if (restUrl != null) {[m
[31m-            restService = JAXRSClientFactory.create(restUrl.buildUrl(), RESTService.class);[m
[31m-            restServiceClient = WebClient.client(restService);[m
[31m-[m
[31m-            if (StringUtils.startsWithIgnoreCase(endpointUrl, "https")) {[m
[31m-                setTLSOptions(openSearchClient);[m
[31m-                setTLSOptions(restServiceClient);[m
[31m-            }[m
[32m+[m[32m            restServiceClient = new SecureCxfClientFactory(restUrl.buildUrl(), RESTService.class,[m
[32m+[m[32m                    username, password);[m
         }[m
     }[m
 [m
     /**[m
      * Generates a DDF REST URL from an OpenSearch URL[m
[32m+[m[32m     *[m
      * @param query[m
      * @param endpointUrl[m
      * @return URL in String format[m
[36m@@ -120,6 +107,7 @@[m [mprivate String createRestUrl(Query query, String endpointUrl, boolean retrieveRe[m
 [m
     /**[m
      * Creates a new RestUrl object based on an OpenSearch URL[m
[32m+[m[32m     *[m
      * @param url[m
      * @return RestUrl object for a DDF REST endpoint[m
      */[m
[36m@@ -138,52 +126,40 @@[m [mprivate RestUrl newRestUrl(String url) {[m
 [m
     /**[m
      * Returns the OpenSearch {@link org.apache.cxf.jaxrs.client.WebClient}[m
[32m+[m[32m     *[m
      * @return {@link org.apache.cxf.jaxrs.client.WebClient}[m
      */[m
[31m-    public WebClient getOpenSearchWebClient() {[m
[31m-        return WebClient.fromClient(openSearchClient);[m
[32m+[m[32m    public WebClient getOpenSearchWebClient(Subject subject) throws SecurityServiceException {[m
[32m+[m[32m        if (subject == null) {[m
[32m+[m[32m            return (WebClient) openSearchClient.getClientForSystem();[m
[32m+[m[32m        } else {[m
[32m+[m[32m            return (WebClient) openSearchClient.getClientForSubject(subject);[m
[32m+[m[32m        }[m
     }[m
 [m
     /**[m
      * Returns the DDF REST {@link org.apache.cxf.jaxrs.client.WebClient}[m
[32m+[m[32m     *[m
      * @return {@link org.apache.cxf.jaxrs.client.WebClient}[m
      */[m
[31m-    public WebClient getRestWebClient() {[m
[32m+[m[32m    public WebClient getRestWebClient(Subject subject) throws SecurityServiceException {[m
         if (restServiceClient != null) {[m
[31m-            return WebClient.fromClient(restServiceClient);[m
[32m+[m[32m            if (subject == null) {[m
[32m+[m[32m                return (WebClient) restServiceClient.getClientForSystem();[m
[32m+[m[32m            } else {[m
[32m+[m[32m                return (WebClient) restServiceClient.getClientForSubject(subject);[m
[32m+[m[32m            }[m
         }[m
         return null;[m
     }[m
 [m
     /**[m
[31m-     * Returns an arbitrary {@link org.apache.cxf.jaxrs.client.WebClient} for any {@link org.apache.cxf.jaxrs.client.Client}[m
[31m-     * @param client {@link org.apache.cxf.jaxrs.client.Client}[m
[31m-     * @return {@link org.apache.cxf.jaxrs.client.WebClient}[m
[31m-     */[m
[31m-    public WebClient getWebClientFromClient(Client client) {[m
[31m-        return WebClient.fromClient(client);[m
[31m-    }[m
[31m-[m
[31m-    /**[m
[31m-     * Creates a new OpenSearch {@link org.apache.cxf.jaxrs.client.Client} based on a String URL[m
[31m-     * @param url[m
[31m-     * @return {@link org.apache.cxf.jaxrs.client.Client}[m
[31m-     */[m
[31m-    public Client newOpenSearchClient(String url) {[m
[31m-        OpenSearch proxy = JAXRSClientFactory.create(url, OpenSearch.class);[m
[31m-        Client tmp = WebClient.client(proxy);[m
[31m-        if (StringUtils.startsWithIgnoreCase(url, "https")) {[m
[31m-            setTLSOptions(tmp);[m
[31m-        }[m
[31m-        return tmp;[m
[31m-    }[m
[31m-[m
[31m-    /**[m
      * Creates a new DDF REST {@link org.apache.cxf.jaxrs.client.Client} based on an OpenSearch[m
      * String URL.[m
[31m-     * @param url - OpenSearch URL[m
[31m-     * @param query - Query to be performed[m
[31m-     * @param metacardId - MetacardId to search for[m
[32m+[m[32m     *[m
[32m+[m[32m     * @param url              - OpenSearch URL[m
[32m+[m[32m     * @param query            - Query to be performed[m
[32m+[m[32m     * @param metacardId       - MetacardId to search for[m
      * @param retrieveResource - true if this is a resource request[m
      * @return {@link org.apache.cxf.jaxrs.client.Client}[m
      */[m
[36m@@ -195,7 +171,7 @@[m [mpublic Client newRestClient(String url, Query query, String metacardId,[m
             RestUrl restUrl = newRestUrl(url);[m
 [m
             if (restUrl != null) {[m
[31m-                if(StringUtils.isNotEmpty(metacardId)) {[m
[32m+[m[32m                if (StringUtils.isNotEmpty(metacardId)) {[m
                     restUrl.setId(metacardId);[m
                 }[m
                 restUrl.setRetrieveResource(retrieveResource);[m
[36m@@ -204,50 +180,11 @@[m [mpublic Client newRestClient(String url, Query query, String metacardId,[m
         }[m
         Client tmp = null;[m
         if (url != null) {[m
[31m-            RESTService proxy = JAXRSClientFactory.create(url, RESTService.class);[m
[31m-            tmp = WebClient.client(proxy);[m
[31m-            if (StringUtils.startsWithIgnoreCase(url, "https")) {[m
[31m-                setTLSOptions(tmp);[m
[32m+[m[32m            try {[m
[32m+[m[32m                tmp = (Client) new SecureCxfClientFactory<RESTService>(url, RESTService.class);[m
[32m+[m[32m            } catch (SecurityServiceException e) {[m
             }[m
         }[m
         return tmp;[m
     }[m
[31m-[m
[31m-    /**[m
[31m-     * Add TLS and Basic Auth credentials to the underlying {@link org.apache.cxf.transport.http.HTTPConduit}[m
[31m-     * @param client[m
[31m-     */[m
[31m-    private void setTLSOptions(Client client) {[m
[31m-        ClientConfiguration clientConfiguration = WebClient.getConfig(client);[m
[31m-[m
[31m-        HTTPConduit httpConduit = clientConfiguration.getHttpConduit();[m
[31m-[m
[31m-        if(StringUtils.isNotEmpty(username) && StringUtils.isNotEmpty(password)) {[m
[31m-            if(httpConduit.getAuthorization() != null) {[m
[31m-                httpConduit.getAuthorization().setUserName(username);[m
[31m-                httpConduit.getAuthorization().setPassword(password);[m
[31m-            }[m
[31m-        }[m
[31m-[m
[31m-        TLSClientParameters tlsParams = securitySettingsService.getTLSParameters();[m
[31m-        tlsParams.setDisableCNCheck(true);[m
[31m-        httpConduit.setTlsClientParameters(tlsParams);[m
[31m-[m
[31m-    }[m
[31m-[m
[31m-    public String getUsername() {[m
[31m-        return username;[m
[31m-    }[m
[31m-[m
[31m-    public void setUsername(String username) {[m
[31m-        this.username = username;[m
[31m-    }[m
[31m-[m
[31m-    public String getPassword() {[m
[31m-        return password;[m
[31m-    }[m
[31m-[m
[31m-    public void setPassword(String password) {[m
[31m-        this.password = password;[m
[31m-    }[m
 }[m
[1mdiff --git opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java[m
[1mindex 209feb4..673cecc 100755[m
[1m--- opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java[m
[1m+++ opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java[m
[36m@@ -14,56 +14,6 @@[m
  **/[m
 package ddf.catalog.source.opensearch;[m
 [m
[31m-import java.io.ByteArrayInputStream;[m
[31m-import java.io.IOException;[m
[31m-import java.io.InputStream;[m
[31m-import java.io.InputStreamReader;[m
[31m-import java.io.Serializable;[m
[31m-import java.net.URI;[m
[31m-import java.nio.charset.StandardCharsets;[m
[31m-import java.util.ArrayList;[m
[31m-import java.util.Arrays;[m
[31m-import java.util.Collection;[m
[31m-import java.util.Collections;[m
[31m-import java.util.Date;[m
[31m-import java.util.HashMap;[m
[31m-import java.util.List;[m
[31m-import java.util.Map;[m
[31m-import java.util.Set;[m
[31m-[m
[31m-import javax.activation.MimeType;[m
[31m-import javax.activation.MimeTypeParseException;[m
[31m-import javax.ws.rs.client.ClientException;[m
[31m-import javax.ws.rs.core.HttpHeaders;[m
[31m-import javax.ws.rs.core.Response;[m
[31m-import javax.xml.namespace.QName;[m
[31m-import javax.xml.transform.TransformerException;[m
[31m-import javax.xml.xpath.XPathFactory;[m
[31m-[m
[31m-import ddf.security.settings.SecuritySettingsService;[m
[31m-import org.apache.abdera.Abdera;[m
[31m-import org.apache.abdera.ext.opensearch.OpenSearchConstants;[m
[31m-import org.apache.abdera.model.Category;[m
[31m-import org.apache.abdera.model.Element;[m
[31m-import org.apache.abdera.model.Entry;[m
[31m-import org.apache.abdera.model.Feed;[m
[31m-import org.apache.abdera.parser.Parser;[m
[31m-import org.apache.commons.io.IOUtils;[m
[31m-import org.apache.commons.lang.StringUtils;[m
[31m-import org.apache.cxf.jaxrs.client.Client;[m
[31m-import org.apache.cxf.jaxrs.client.WebClient;[m
[31m-import org.codice.ddf.configuration.ConfigurationWatcher;[m
[31m-import org.codice.ddf.security.common.jaxrs.RestSecurity;[m
[31m-import org.geotools.filter.FilterTransformer;[m
[31m-import org.slf4j.Logger;[m
[31m-import org.slf4j.LoggerFactory;[m
[31m-[m
[31m-import java.io.StringReader;[m
[31m-[m
[31m-import javax.xml.xpath.XPath;[m
[31m-import javax.xml.xpath.XPathExpressionException;[m
[31m-import org.xml.sax.InputSource;[m
[31m-[m
 import ddf.catalog.data.ContentType;[m
 import ddf.catalog.data.Metacard;[m
 import ddf.catalog.data.Result;[m
[36m@@ -90,36 +40,70 @@[m
 import ddf.catalog.transform.InputTransformer;[m
 import ddf.security.SecurityConstants;[m
 import ddf.security.Subject;[m
[32m+[m[32mimport ddf.security.service.SecurityServiceException;[m
[32m+[m[32mimport org.apache.abdera.Abdera;[m
[32m+[m[32mimport org.apache.abdera.ext.opensearch.OpenSearchConstants;[m
[32m+[m[32mimport org.apache.abdera.model.Category;[m
[32m+[m[32mimport org.apache.abdera.model.Element;[m
[32m+[m[32mimport org.apache.abdera.model.Entry;[m
[32m+[m[32mimport org.apache.abdera.model.Feed;[m
[32m+[m[32mimport org.apache.abdera.parser.Parser;[m
[32m+[m[32mimport org.apache.commons.io.IOUtils;[m
[32m+[m[32mimport org.apache.commons.lang.StringUtils;[m
[32m+[m[32mimport org.apache.cxf.jaxrs.client.Client;[m
[32m+[m[32mimport org.apache.cxf.jaxrs.client.WebClient;[m
[32m+[m[32mimport org.geotools.filter.FilterTransformer;[m
[32m+[m[32mimport org.slf4j.Logger;[m
[32m+[m[32mimport org.slf4j.LoggerFactory;[m
[32m+[m[32mimport org.xml.sax.InputSource;[m
[32m+[m
[32m+[m[32mimport javax.activation.MimeType;[m
[32m+[m[32mimport javax.activation.MimeTypeParseException;[m
[32m+[m[32mimport javax.ws.rs.client.ClientException;[m
[32m+[m[32mimport javax.ws.rs.core.HttpHeaders;[m
[32m+[m[32mimport javax.ws.rs.core.Response;[m
[32m+[m[32mimport javax.xml.namespace.QName;[m
[32m+[m[32mimport javax.xml.transform.TransformerException;[m
[32m+[m[32mimport javax.xml.xpath.XPath;[m
[32m+[m[32mimport javax.xml.xpath.XPathExpressionException;[m
[32m+[m[32mimport javax.xml.xpath.XPathFactory;[m
[32m+[m[32mimport java.io.ByteArrayInputStream;[m
[32m+[m[32mimport java.io.IOException;[m
[32m+[m[32mimport java.io.InputStream;[m
[32m+[m[32mimport java.io.InputStreamReader;[m
[32m+[m[32mimport java.io.Serializable;[m
[32m+[m[32mimport java.io.StringReader;[m
[32m+[m[32mimport java.net.URI;[m
[32m+[m[32mimport java.nio.charset.StandardCharsets;[m
[32m+[m[32mimport java.util.ArrayList;[m
[32m+[m[32mimport java.util.Arrays;[m
[32m+[m[32mimport java.util.Collection;[m
[32m+[m[32mimport java.util.Collections;[m
[32m+[m[32mimport java.util.Date;[m
[32m+[m[32mimport java.util.HashMap;[m
[32m+[m[32mimport java.util.List;[m
[32m+[m[32mimport java.util.Map;[m
[32m+[m[32mimport java.util.Set;[m
 [m
 /**[m
  * Federated site that talks via OpenSearch to the DDF platform. Communication is usually performed[m
  * via https which requires a keystore and trust store to be provided.[m
[31m- *[m
  */[m
 public final class OpenSearchSource implements FederatedSource, ConfiguredService {[m
 [m
[31m-    static final String BAD_URL_MESSAGE = "Bad url given for remote source";[m
[31m-[m
[31m-    static final String COULD_NOT_RETRIEVE_RESOURCE_MESSAGE = "Could not retrieve resource";[m
[31m-[m
     public static final String NAME = "name";[m
 [m
[31m-    private boolean isInitialized = false;[m
[31m-[m
[31m-    // service properties[m
[31m-    private String shortname;[m
[31m-[m
[31m-    private boolean lastAvailable;[m
[32m+[m[32m    public static final String BYTES_TO_SKIP = "BytesToSkip";[m
 [m
[31m-    private Date lastAvailableDate = null;[m
[32m+[m[32m    public static final String BYTES_SKIPPED = "BytesSkipped";[m
 [m
[31m-    private boolean localQueryOnly;[m
[32m+[m[32m    static final String BAD_URL_MESSAGE = "Bad url given for remote source";[m
 [m
[31m-    private boolean shouldConvertToBBox;[m
[32m+[m[32m    static final String COULD_NOT_RETRIEVE_RESOURCE_MESSAGE = "Could not retrieve resource";[m
 [m
[31m-    private String endpointUrl;[m
[32m+[m[32m    static final String HEADER_ACCEPT_RANGES = "Accept-Ranges";[m
 [m
[31m-    private InputTransformer inputTransformer;[m
[32m+[m[32m    static final String BYTES = "bytes";[m
 [m
     private static final String ORGANIZATION = "DDF";[m
 [m
[36m@@ -133,30 +117,37 @@[m
 [m
     private static final String LOCAL_SEARCH_PARAMETER = "local";[m
 [m
[31m-    static final String HEADER_ACCEPT_RANGES = "Accept-Ranges";[m
[31m-[m
     private static final String HEADER_RANGE = "Range";[m
 [m
[31m-    public static final String BYTES_TO_SKIP = "BytesToSkip";[m
[31m-[m
[31m-    public static final String BYTES_SKIPPED = "BytesSkipped";[m
[31m-[m
[31m-    static final String BYTES = "bytes";[m
[31m-[m
     private static final String BYTES_EQUAL = "bytes=";[m
 [m
     private static final Logger LOGGER = LoggerFactory.getLogger(OpenSearchSource.class);[m
 [m
[31m-    private FilterAdapter filterAdapter;[m
[31m-[m
     // expensive creation, meant to be done once[m
     private static final Abdera ABDERA = new Abdera();[m
 [m
[31m-    private String configurationPid;[m
[31m-[m
     protected OpenSearchConnection openSearchConnection;[m
 [m
[31m-    private SecuritySettingsService securitySettingsService;[m
[32m+[m[32m    private boolean isInitialized = false;[m
[32m+[m
[32m+[m[32m    // service properties[m
[32m+[m[32m    private String shortname;[m
[32m+[m
[32m+[m[32m    private boolean lastAvailable;[m
[32m+[m
[32m+[m[32m    private Date lastAvailableDate = null;[m
[32m+[m
[32m+[m[32m    private boolean localQueryOnly;[m
[32m+[m
[32m+[m[32m    private boolean shouldConvertToBBox;[m
[32m+[m
[32m+[m[32m    private String endpointUrl;[m
[32m+[m
[32m+[m[32m    private InputTransformer inputTransformer;[m
[32m+[m
[32m+[m[32m    private FilterAdapter filterAdapter;[m
[32m+[m
[32m+[m[32m    private String configurationPid;[m
 [m
     private List<String> parameters;[m
 [m
[36m@@ -169,8 +160,6 @@[m
     /**[m
      * Creates an OpenSearch Site instance. Sets an initial default endpointUrl that can be[m
      * overwritten using the setter methods.[m
[31m-     *[m
[31m-     * @throws ddf.catalog.source.UnsupportedQueryException[m
      */[m
     public OpenSearchSource(FilterAdapter filterAdapter) {[m
         this.filterAdapter = filterAdapter;[m
[36m@@ -180,7 +169,7 @@[m [mpublic OpenSearchSource(FilterAdapter filterAdapter) {[m
      * Called when this OpenSearch Source is created, but after all of the setter methods have been[m
      * called for each property specified in the metatype.xml file.[m
      */[m
[31m-    public void init() {[m
[32m+[m[32m    public void init() throws SecurityServiceException {[m
         isInitialized = true;[m
         configureClient();[m
     }[m
[36m@@ -189,30 +178,38 @@[m [mpublic void destroy() {[m
         LOGGER.info("Nothing to destroy.");[m
     }[m
 [m
[31m-    protected void configureClient() {[m
[31m-        openSearchConnection = new OpenSearchConnection(endpointUrl, filterAdapter, securitySettingsService, username, password);[m
[32m+[m[32m    protected void configureClient() throws SecurityServiceException {[m
[32m+[m[32m        openSearchConnection = new OpenSearchConnection(endpointUrl, filterAdapter, username,[m
[32m+[m[32m                password);[m
     }[m
 [m
     @Override[m
     public boolean isAvailable() {[m
         boolean isAvailable = false;[m
[31m-        if (!lastAvailable[m
[31m-                || (lastAvailableDate.before(new Date(System.currentTimeMillis()[m
[31m-                        - AVAILABLE_TIMEOUT_CHECK)))) {[m
[32m+[m[32m        if (!lastAvailable || (lastAvailableDate[m
[32m+[m[32m                .before(new Date(System.currentTimeMillis() - AVAILABLE_TIMEOUT_CHECK)))) {[m
 [m
[31m-            WebClient client = openSearchConnection.getOpenSearchWebClient();[m
[31m-[m
[31m-            Response response = null;[m
[32m+[m[32m            boolean failed = false;[m
[32m+[m[32m            WebClient client = null;[m
             try {[m
[31m-                response = client.head();[m
[31m-            } catch (ClientException e) {[m
[31m-                LOGGER.warn("Web Client was unable to connect to endpoint.", e);[m
[32m+[m[32m                client = openSearchConnection.getOpenSearchWebClient(null);[m
[32m+[m[32m            } catch (SecurityServiceException e) {[m
[32m+[m[32m                failed = true;[m
             }[m
 [m
[31m-            if(response != null && !(response.getStatus() >= 404 || response.getStatus() == 400[m
[31m-                    || response.getStatus() == 402)) {[m
[31m-                isAvailable = true;[m
[31m-                lastAvailableDate = new Date();[m
[32m+[m[32m            if (!failed) {[m
[32m+[m[32m                Response response = null;[m
[32m+[m[32m                try {[m
[32m+[m[32m                    response = client.head();[m
[32m+[m[32m                } catch (ClientException e) {[m
[32m+[m[32m                    LOGGER.warn("Web Client was unable to connect to endpoint.", e);[m
[32m+[m[32m                }[m
[32m+[m
[32m+[m[32m                if (response != null && !(response.getStatus() >= 404 || response.getStatus() == 400[m
[32m+[m[32m                        || response.getStatus() == 402)) {[m
[32m+[m[32m                    isAvailable = true;[m
[32m+[m[32m                    lastAvailableDate = new Date();[m
[32m+[m[32m                }[m
             }[m
         } else {[m
             isAvailable = lastAvailable;[m
[36m@@ -223,7 +220,7 @@[m [mpublic boolean isAvailable() {[m
 [m
     @Override[m
     public boolean isAvailable(SourceMonitor callback) {[m
[31m-        if(isAvailable()) {[m
[32m+[m[32m        if (isAvailable()) {[m
             callback.setAvailable();[m
             return true;[m
         } else {[m
[36m@@ -240,14 +237,18 @@[m [mpublic SourceResponse query(QueryRequest queryRequest) throws UnsupportedQueryEx[m
         Serializable metacardId = queryRequest.getPropertyValue(Metacard.ID);[m
         SourceResponseImpl response = null;[m
 [m
[31m-        WebClient openSearchWebClient = openSearchConnection.getOpenSearchWebClient();[m
[31m-[m
         Subject subject = null;[m
         if (queryRequest.hasProperties()) {[m
             Object subjectObj = queryRequest.getProperties()[m
                     .get(SecurityConstants.SECURITY_SUBJECT);[m
             subject = (Subject) subjectObj;[m
[31m-            RestSecurity.setSubjectOnClient(subject, openSearchWebClient);[m
[32m+[m[32m        }[m
[32m+[m
[32m+[m[32m        WebClient openSearchWebClient;[m
[32m+[m[32m        try {[m
[32m+[m[32m            openSearchWebClient = openSearchConnection.getOpenSearchWebClient(subject);[m
[32m+[m[32m        } catch (SecurityServiceException e) {[m
[32m+[m[32m            throw new UnsupportedQueryException(e);[m
         }[m
 [m
         Query query = queryRequest.getQuery();[m
[36m@@ -258,40 +259,42 @@[m [mpublic SourceResponse query(QueryRequest queryRequest) throws UnsupportedQueryEx[m
 [m
         boolean canDoOpenSearch = setOpenSearchParameters(query, subject, openSearchWebClient);[m
 [m
[31m-        if(canDoOpenSearch) {[m
[32m+[m[32m        if (canDoOpenSearch) {[m
 [m
             InputStream responseStream = performRequest(openSearchWebClient);[m
 [m
[31m-            response = new SourceResponseImpl(queryRequest,[m
[31m-                    new ArrayList<Result>());[m
[32m+[m[32m            response = new SourceResponseImpl(queryRequest, new ArrayList<Result>());[m
 [m
             if (responseStream != null) {[m
                 response = processResponse(responseStream, queryRequest);[m
             }[m
         } else {[m
[31m-            Client restClient = openSearchConnection.newRestClient(endpointUrl,[m
[31m-                    queryRequest.getQuery(),[m
[31m-                    (String) metacardId, false);[m
[32m+[m[32m            Client restClient = openSearchConnection[m
[32m+[m[32m                    .newRestClient(endpointUrl, queryRequest.getQuery(), (String) metacardId,[m
[32m+[m[32m                            false);[m
 [m
             if (restClient != null) {[m
[31m-                WebClient restWebClient = openSearchConnection.getWebClientFromClient(restClient);[m
[31m-[m
[31m-                if (queryRequest.hasProperties()) {[m
[31m-                    Object subjectObj = queryRequest.getProperties()[m
[31m-                            .get(SecurityConstants.SECURITY_SUBJECT);[m
[31m-                    subject = (Subject) subjectObj;[m
[31m-                    RestSecurity.setSubjectOnClient(subject, restWebClient);[m
[32m+[m[32m                WebClient restWebClient;[m
[32m+[m[32m                try {[m
[32m+[m[32m                    if (queryRequest.hasProperties()) {[m
[32m+[m[32m                        Object subjectObj = queryRequest.getProperties()[m
[32m+[m[32m                                .get(SecurityConstants.SECURITY_SUBJECT);[m
[32m+[m[32m                        subject = (Subject) subjectObj;[m
[32m+[m[32m                        restWebClient = openSearchConnection.getRestWebClient(subject);[m
[32m+[m[32m                    } else {[m
[32m+[m[32m                        restWebClient = openSearchConnection.getRestWebClient(null);[m
[32m+[m[32m                    }[m
[32m+[m[32m                } catch (SecurityServiceException e) {[m
[32m+[m[32m                    throw new UnsupportedQueryException(e);[m
                 }[m
 [m
                 InputStream responseStream = performRequest(restWebClient);[m
 [m
                 Metacard metacard = null;[m
[31m-                List<Result> resultQueue = new ArrayList<Result>();[m
[32m+[m[32m                List<Result> resultQueue = new ArrayList<>();[m
                 try {[m
                     metacard = inputTransformer.transform(responseStream);[m
[31m-                } catch (IOException e) {[m
[31m-                    LOGGER.debug("Problem with transformation.", e);[m
[31m-                } catch (CatalogTransformerException e) {[m
[32m+[m[32m                } catch (IOException | CatalogTransformerException e) {[m
                     LOGGER.debug("Problem with transformation.", e);[m
                 }[m
                 if (metacard != null) {[m
[36m@@ -322,7 +325,7 @@[m [mprivate InputStream performRequest(WebClient client) throws UnsupportedQueryExce[m
         InputStream stream = null;[m
         Object entityObj = clientResponse.getEntity();[m
         if (entityObj != null) {[m
[31m-            stream = (InputStream)entityObj;[m
[32m+[m[32m            stream = (InputStream) entityObj;[m
         }[m
         if (Response.Status.OK.getStatusCode() != clientResponse.getStatus()) {[m
             String error = "";[m
[36m@@ -331,11 +334,11 @@[m [mprivate InputStream performRequest(WebClient client) throws UnsupportedQueryExce[m
                     error = IOUtils.toString(stream);[m
                 }[m
             } catch (IOException ioe) {[m
[31m-                LOGGER.debug("Could not convert error message to a string for output.",[m
[31m-                        ioe);[m
[32m+[m[32m                LOGGER.debug("Could not convert error message to a string for output.", ioe);[m
             }[m
[31m-            String errorMsg = "Received error code from remote source (status "[m
[31m-                    + clientResponse.getStatus() + "): " + error;[m
[32m+[m[32m            String errorMsg =[m
[32m+[m[32m                    "Received error code from remote source (status " + clientResponse.getStatus()[m
[32m+[m[32m                            + "): " + error;[m
             LOGGER.warn(errorMsg);[m
             throw new UnsupportedQueryException(errorMsg);[m
         }[m
[36m@@ -361,7 +364,8 @@[m [mprotected boolean setOpenSearchParameters(Query query, Subject subject, WebClien[m
         ContextualSearch contextualFilter = visitor.getContextualSearch();[m
 [m
         //TODO fix this so we aren't just triggering off of a contextual query[m
[31m-        if(contextualFilter != null && StringUtils.isNotEmpty(contextualFilter.getSearchPhrase())) {[m
[32m+[m[32m        if (contextualFilter != null && StringUtils[m
[32m+[m[32m                .isNotEmpty(contextualFilter.getSearchPhrase())) {[m
             // All queries must have at least a search phrase to be valid, hence this check[m
             // for a contextual filter with a non-empty search phrase[m
             OpenSearchSiteUtil.populateSearchOptions(client, query, subject, parameters);[m
[36m@@ -381,16 +385,17 @@[m [mprotected boolean setOpenSearchParameters(Query query, Subject subject, WebClien[m
             if (spatialFilter != null) {[m
                 if (spatialFilter instanceof SpatialDistanceFilter) {[m
                     try {[m
[31m-                        OpenSearchSiteUtil.populateGeospatial(client,[m
[31m-                                (SpatialDistanceFilter) spatialFilter, shouldConvertToBBox,[m
[31m-                                parameters);[m
[32m+[m[32m                        OpenSearchSiteUtil[m
[32m+[m[32m                                .populateGeospatial(client, (SpatialDistanceFilter) spatialFilter,[m
[32m+[m[32m                                        shouldConvertToBBox, parameters);[m
                     } catch (UnsupportedQueryException e) {[m
                         LOGGER.info("Problem with populating geospatial criteria. ", e);[m
                     }[m
                 } else {[m
                     try {[m
[31m-                        OpenSearchSiteUtil.populateGeospatial(client, spatialFilter,[m
[31m-                                shouldConvertToBBox, parameters);[m
[32m+[m[32m                        OpenSearchSiteUtil[m
[32m+[m[32m                                .populateGeospatial(client, spatialFilter, shouldConvertToBBox,[m
[32m+[m[32m                                        parameters);[m
                     } catch (UnsupportedQueryException e) {[m
                         LOGGER.info("Problem with populating geospatial criteria. ", e);[m
                     }[m
[36m@@ -413,11 +418,12 @@[m [mprotected boolean setOpenSearchParameters(Query query, Subject subject, WebClien[m
      * @return[m
      * @throws ddf.catalog.source.UnsupportedQueryException[m
      */[m
[31m-    private SourceResponseImpl processResponse(InputStream is, QueryRequest queryRequest) throws UnsupportedQueryException {[m
[31m-        List<Result> resultQueue = new ArrayList<Result>();[m
[32m+[m[32m    private SourceResponseImpl processResponse(InputStream is, QueryRequest queryRequest)[m
[32m+[m[32m            throws UnsupportedQueryException {[m
[32m+[m[32m        List<Result> resultQueue = new ArrayList<>();[m
 [m
         ClassLoader tccl = Thread.currentThread().getContextClassLoader();[m
[31m-        Parser parser = null;[m
[32m+[m[32m        Parser parser;[m
         org.apache.abdera.model.Document<Feed> atomDoc;[m
         try {[m
 [m
[36m@@ -439,8 +445,8 @@[m [mprivate SourceResponseImpl processResponse(InputStream is, QueryRequest queryReq[m
         long totalResults = entries.size();[m
 [m
         // OSGi has some weird issues with Abdera's XPath, so just traverse down the element tree[m
[31m-        Element totalResultsElement = atomDoc.getRoot().getExtension([m
[31m-                OpenSearchConstants.TOTAL_RESULTS);[m
[32m+[m[32m        Element totalResultsElement = atomDoc.getRoot()[m
[32m+[m[32m                .getExtension(OpenSearchConstants.TOTAL_RESULTS);[m
 [m
         if (totalResultsElement != null) {[m
             try {[m
[36m@@ -462,8 +468,7 @@[m [mprivate SourceResponseImpl processResponse(InputStream is, QueryRequest queryReq[m
      * Creates a single response from input parameters. Performs XPath operations on the document to[m
      * retrieve data not passed in.[m
      *[m
[31m-     * @param entry[m
[31m-     *            a single Atom entry[m
[32m+[m[32m     * @param entry a single Atom entry[m
      * @return single response[m
      * @throws ddf.catalog.source.UnsupportedQueryException[m
      */[m
[36m@@ -530,8 +535,8 @@[m [mprivate Result createResponseFromEntry(Entry entry) throws UnsupportedQueryExcep[m
             ClassLoader tccl2 = Thread.currentThread().getContextClassLoader();[m
             try {[m
 [m
[31m-                Thread.currentThread().setContextClassLoader([m
[31m-                        OpenSearchSource.class.getClassLoader());[m
[32m+[m[32m                Thread.currentThread()[m
[32m+[m[32m                        .setContextClassLoader(OpenSearchSource.class.getClassLoader());[m
                 List<Category> categories = entry.getCategories();[m
                 if (!categories.isEmpty() && categories.get(0) != null) {[m
                     String term = categories.get(0).toString();[m
[36m@@ -541,7 +546,8 @@[m [mprivate Result createResponseFromEntry(Entry entry) throws UnsupportedQueryExcep[m
                     XPath xpath = XPathFactory.newInstance().newXPath();[m
                     InputSource inputSource = new InputSource(new StringReader(term));[m
                     try {[m
[31m-                        contentType = xpath.evaluate("//*[local-name()='category']/@term", inputSource);[m
[32m+[m[32m                        contentType = xpath[m
[32m+[m[32m                                .evaluate("//*[local-name()='category']/@term", inputSource);[m
                         metacard.setContentTypeName(contentType);[m
                     } catch (XPathExpressionException e) {[m
                         LOGGER.info("Unable to parse categories for contentType");[m
[36m@@ -581,18 +587,15 @@[m [mprivate MetacardImpl getMetacardImpl(Metacard oldMetacard) {[m
 [m
     // Update the WebClient with a Range header instructing the endpoint to skip bytesToSkip bytes.[m
     private void constructRangeHeader(WebClient webClient, Long bytesToSkip) {[m
[31m-        StringBuilder headerValue = new StringBuilder(BYTES_EQUAL);[m
[31m-        headerValue.append(bytesToSkip.toString());[m
[31m-        headerValue.append("-");[m
 [m
[31m-        webClient.header(HEADER_RANGE, headerValue.toString());[m
[32m+[m[32m        webClient.header(HEADER_RANGE, BYTES_EQUAL + bytesToSkip.toString() + "-");[m
     }[m
 [m
     private Metacard parseContent(String content, String id) {[m
         if (inputTransformer != null && content != null && !content.isEmpty()) {[m
             try {[m
[31m-                return inputTransformer.transform(new ByteArrayInputStream(content.getBytes([m
[31m-                        StandardCharsets.UTF_8)), id);[m
[32m+[m[32m                return inputTransformer.transform([m
[32m+[m[32m                        new ByteArrayInputStream(content.getBytes(StandardCharsets.UTF_8)), id);[m
             } catch (IOException e) {[m
                 LOGGER.warn("Unable to read metacard content from Atom feed.", e);[m
             } catch (CatalogTransformerException e) {[m
[36m@@ -605,20 +608,6 @@[m [mprivate Metacard parseContent(String content, String id) {[m
     }[m
 [m
     /**[m
[31m-     * Set URL of the endpoint.[m
[31m-     *[m
[31m-     * @param endpointUrl[m
[31m-     *            Full url of the endpoint.[m
[31m-     */[m
[31m-    public void setEndpointUrl(String endpointUrl) {[m
[31m-        this.endpointUrl = endpointUrl;[m
[31m-[m
[31m-        if(isInitialized) {[m
[31m-            configureClient();[m
[31m-        }[m
[31m-    }[m
[31m-[m
[31m-    /**[m
      * Get the URL of the endpoint.[m
      *[m
      * @return[m
[36m@@ -628,6 +617,19 @@[m [mpublic String getEndpointUrl() {[m
         return endpointUrl;[m
     }[m
 [m
[32m+[m[32m    /**[m
[32m+[m[32m     * Set URL of the endpoint.[m
[32m+[m[32m     *[m
[32m+[m[32m     * @param endpointUrl Full url of the endpoint.[m
[32m+[m[32m     */[m
[32m+[m[32m    public void setEndpointUrl(String endpointUrl) throws SecurityServiceException {[m
[32m+[m[32m        this.endpointUrl = endpointUrl;[m
[32m+[m
[32m+[m[32m        if (isInitialized) {[m
[32m+[m[32m            configureClient();[m
[32m+[m[32m        }[m
[32m+[m[32m    }[m
[32m+[m
     @Override[m
     public String getDescription() {[m
         return DESCRIPTION;[m
[36m@@ -647,8 +649,7 @@[m [mpublic String getId() {[m
      * Sets the shortname for this site. This shortname is used to identify the site when performing[m
      * federated queries.[m
      *[m
[31m-     * @param shortname[m
[31m-     *            Name of this site.[m
[32m+[m[32m     * @param shortname Name of this site.[m
      */[m
     public void setShortname(String shortname) {[m
         this.shortname = shortname;[m
[36m@@ -664,24 +665,12 @@[m [mpublic String getVersion() {[m
         return "2.0";[m
     }[m
 [m
[31m-    public void setInputTransformer(InputTransformer inputTransformer) {[m
[31m-        this.inputTransformer = inputTransformer;[m
[31m-    }[m
[31m-[m
     public InputTransformer getInputTransformer() {[m
         return inputTransformer;[m
     }[m
 [m
[31m-    /**[m
[31m-     * Sets the boolean flag that indicates all queries executed should be to its local source only,[m
[31m-     * i.e., no federated or enterprise queries.[m
[31m-     *[m
[31m-     * @param localQueryOnly[m
[31m-     *            true indicates only local queries, false indicates enterprise query[m
[31m-     */[m
[31m-    public void setLocalQueryOnly(boolean localQueryOnly) {[m
[31m-        LOGGER.trace("Setting localQueryOnly = {}", localQueryOnly);[m
[31m-        this.localQueryOnly = localQueryOnly;[m
[32m+[m[32m    public void setInputTransformer(InputTransformer inputTransformer) {[m
[32m+[m[32m        this.inputTransformer = inputTransformer;[m
     }[m
 [m
     /**[m
[36m@@ -695,13 +684,14 @@[m [mpublic boolean getLocalQueryOnly() {[m
     }[m
 [m
     /**[m
[31m-     * Sets the boolean flag that tells the code to convert point-radius and polygon geometries to a[m
[31m-     * bounding box before sending them.[m
[32m+[m[32m     * Sets the boolean flag that indicates all queries executed should be to its local source only,[m
[32m+[m[32m     * i.e., no federated or enterprise queries.[m
      *[m
[31m-     * @param shouldConvertToBBox[m
[32m+[m[32m     * @param localQueryOnly true indicates only local queries, false indicates enterprise query[m
      */[m
[31m-    public void setShouldConvertToBBox(boolean shouldConvertToBBox) {[m
[31m-        this.shouldConvertToBBox = shouldConvertToBBox;[m
[32m+[m[32m    public void setLocalQueryOnly(boolean localQueryOnly) {[m
[32m+[m[32m        LOGGER.trace("Setting localQueryOnly = {}", localQueryOnly);[m
[32m+[m[32m        this.localQueryOnly = localQueryOnly;[m
     }[m
 [m
     /**[m
[36m@@ -714,16 +704,27 @@[m [mpublic boolean getShouldConvertToBBox() {[m
         return shouldConvertToBBox;[m
     }[m
 [m
[32m+[m[32m    /**[m
[32m+[m[32m     * Sets the boolean flag that tells the code to convert point-radius and polygon geometries to a[m
[32m+[m[32m     * bounding box before sending them.[m
[32m+[m[32m     *[m
[32m+[m[32m     * @param shouldConvertToBBox[m
[32m+[m[32m     */[m
[32m+[m[32m    public void setShouldConvertToBBox(boolean shouldConvertToBBox) {[m
[32m+[m[32m        this.shouldConvertToBBox = shouldConvertToBBox;[m
[32m+[m[32m    }[m
[32m+[m
     @Override[m
     public ResourceResponse retrieveResource(URI uri, Map<String, Serializable> requestProperties)[m
             throws ResourceNotFoundException, ResourceNotSupportedException, IOException {[m
 [m
[31m-        Long bytesToSkip = Long.valueOf(0);[m
[32m+[m[32m        Long bytesToSkip = (long) 0;[m
         final String methodName = "retrieveResource";[m
         LOGGER.trace("ENTRY: {}", methodName);[m
 [m
         if (requestProperties == null) {[m
[31m-            throw new ResourceNotFoundException("Could not retrieve resource with null properties.");[m
[32m+[m[32m            throw new ResourceNotFoundException([m
[32m+[m[32m                    "Could not retrieve resource with null properties.");[m
         }[m
 [m
         Serializable serializableId = requestProperties.get(Metacard.ID);[m
[36m@@ -732,27 +733,29 @@[m [mpublic ResourceResponse retrieveResource(URI uri, Map<String, Serializable> requ[m
 [m
         if (serializableId != null) {[m
             String metacardId = serializableId.toString();[m
[31m-            Client restClient = openSearchConnection.newRestClient(endpointUrl, null, metacardId, true);[m
[32m+[m[32m            Client restClient = openSearchConnection[m
[32m+[m[32m                    .newRestClient(endpointUrl, null, metacardId, true);[m
 [m
             if (restClient != null) {[m
[31m-                Object binaryContent = null;[m
[32m+[m[32m                Object binaryContent;[m
                 MimeType mimeType = null;[m
 [m
[31m-                WebClient webClient = openSearchConnection.getWebClientFromClient(restClient);[m
[32m+[m[32m                WebClient webClient;[m
[32m+[m[32m                try {[m
[32m+[m[32m                    webClient = openSearchConnection.getRestWebClient(subject);[m
[32m+[m[32m                } catch (SecurityServiceException e) {[m
[32m+[m[32m                    throw new IOException(e);[m
[32m+[m[32m                }[m
 [m
                 // If a bytesToSkip property is present add range header[m
[31m-                Map<String, Serializable> responseProperties = new HashMap<String, Serializable>();[m
[32m+[m[32m                Map<String, Serializable> responseProperties = new HashMap<>();[m
                 if (requestProperties.containsKey(BYTES_TO_SKIP)) {[m
                     bytesToSkip = (Long) requestProperties.get(BYTES_TO_SKIP);[m
                     LOGGER.debug("Setting Range header with bytes to skip: {}", bytesToSkip);[m
                     constructRangeHeader(webClient, bytesToSkip);[m
                 }[m
 [m
[31m-                if(subject != null) {[m
[31m-                    RestSecurity.setSubjectOnClient(subject, webClient);[m
[31m-                }[m
[31m-[m
[31m-                Response clientResponse = null;[m
[32m+[m[32m                Response clientResponse;[m
                 try {[m
                     clientResponse = webClient.get();[m
                 } catch (Exception e) {[m
[36m@@ -769,10 +772,8 @@[m [mpublic ResourceResponse retrieveResource(URI uri, Map<String, Serializable> requ[m
                 try {[m
                     mimeType = new MimeType("application/octet-stream");[m
                     String content = null;[m
[31m-                    if(contentType != null) {[m
[31m-                        if(contentType instanceof String) {[m
[31m-                            content = (String) contentType;[m
[31m-                        } else if(contentType instanceof Collection && ((Collection) contentType).size() > 0) {[m
[32m+[m[32m                    if (contentType != null) {[m
[32m+[m[32m                        if (((Collection) contentType).size() > 0) {[m
                             content = (String) ((Collection) contentType).iterator().next();[m
                         }[m
                     }[m
[36m@@ -796,7 +797,8 @@[m [mpublic ResourceResponse retrieveResource(URI uri, Map<String, Serializable> requ[m
                         //DDF-643: Set response property indicating remote JSON Source did the byte skipping[m
                         // so that Catalog Framework's download manager will not try to also skip bytes.[m
                         if ((rangeHeader != null) && (rangeHeader.equals(BYTES))) {[m
[31m-                            LOGGER.info("Adding {} to response properties with value = {}", BYTES_SKIPPED, true);[m
[32m+[m[32m                            LOGGER.info("Adding {} to response properties with value = {}",[m
[32m+[m[32m                                    BYTES_SKIPPED, true);[m
                             responseProperties.put(BYTES_SKIPPED, true);[m
                         }[m
                     }[m
[36m@@ -804,9 +806,9 @@[m [mpublic ResourceResponse retrieveResource(URI uri, Map<String, Serializable> requ[m
                     LOGGER.trace("EXIT: {}", methodName);[m
 [m
                     //DDF-643[m
[31m-                    ResourceResponseImpl resourceResponse = new ResourceResponseImpl(new ResourceImpl([m
[31m-                            (InputStream) binaryContent, mimeType, getId()[m
[31m-                                    + "_Resource_Retrieval:" + System.currentTimeMillis()));[m
[32m+[m[32m                    ResourceResponseImpl resourceResponse = new ResourceResponseImpl([m
[32m+[m[32m                            new ResourceImpl((InputStream) binaryContent, mimeType,[m
[32m+[m[32m                                    getId() + "_Resource_Retrieval:" + System.currentTimeMillis()));[m
                     resourceResponse.setProperties(responseProperties);[m
                     return resourceResponse;[m
                 }[m
[36m@@ -830,19 +832,18 @@[m [mpublic ResourceResponse retrieveResource(URI uri, Map<String, Serializable> requ[m
     @Override[m
     public Set<String> getOptions(Metacard metacard) {[m
         LOGGER.trace("ENTERING/EXITING: getOptions");[m
[31m-        LOGGER.debug("OpenSearch Source \"{}\" does not support resource retrieval options.", getId());[m
[32m+[m[32m        LOGGER.debug("OpenSearch Source \"{}\" does not support resource retrieval options.",[m
[32m+[m[32m                getId());[m
         return Collections.emptySet();[m
     }[m
 [m
     @Override[m
[31m-    public String getConfigurationPid()[m
[31m-    {[m
[32m+[m[32m    public String getConfigurationPid() {[m
         return configurationPid;[m
     }[m
 [m
     @Override[m
[31m-    public void setConfigurationPid(String configurationPid)[m
[31m-    {[m
[32m+[m[32m    public void setConfigurationPid(String configurationPid) {[m
         this.configurationPid = configurationPid;[m
     }[m
 [m
[36m@@ -850,6 +851,10 @@[m [mpublic void setConfigurationPid(String configurationPid)[m
         return parameters;[m
     }[m
 [m
[32m+[m[32m    public void setParameters(String parameters) {[m
[32m+[m[32m        this.parameters = Arrays.asList(parameters.split(","));[m
[32m+[m[32m    }[m
[32m+[m
     public void setParameters(List<String> parameters) {[m
         // workaround for KARAF-1701[m
         if (parameters.size() == 1 && parameters.get(0).contains(",")) {[m
[36m@@ -859,10 +864,6 @@[m [mpublic void setParameters(List<String> parameters) {[m
         }[m
     }[m
 [m
[31m-    public void setParameters(String parameters) {[m
[31m-        this.parameters = Arrays.asList(parameters.split(","));[m
[31m-    }[m
[31m-[m
     public String getUsername() {[m
         return username;[m
     }[m
[36m@@ -880,16 +881,6 @@[m [mpublic void setPassword(String password) {[m
     }[m
 [m
     /**[m
[31m-     * Sets the receive timeout for messages sent from this provider.[m
[31m-     *[m
[31m-     * @param receiveTimeout timeout in milliseconds. 0 sets it to NOT timeout.[m
[31m-     */[m
[31m-    public void setReceiveTimeout(long receiveTimeout) {[m
[31m-        LOGGER.debug("Setting timeout to {}", receiveTimeout);[m
[31m-        this.receiveTimeout = receiveTimeout;[m
[31m-    }[m
[31m-[m
[31m-    /**[m
      * Gets the receive timeout that is being used for current messages.[m
      *[m
      * @return the timeout in milliseconds.[m
[36m@@ -898,7 +889,14 @@[m [mpublic long getReceiveTimeout() {[m
         return this.receiveTimeout;[m
     }[m
 [m
[31m-    public void setSecuritySettings(SecuritySettingsService settingsService) {[m
[31m-        this.securitySettingsService = settingsService;[m
[32m+[m[32m    /**[m
[32m+[m[32m     * Sets the receive timeout for messages sent from this provider.[m
[32m+[m[32m     *[m
[32m+[m[32m     * @param receiveTimeout timeout in milliseconds. 0 sets it to NOT timeout.[m
[32m+[m[32m     */[m
[32m+[m[32m    public void setReceiveTimeout(long receiveTimeout) {[m
[32m+[m[32m        LOGGER.debug("Setting timeout to {}", receiveTimeout);[m
[32m+[m[32m        this.receiveTimeout = receiveTimeout;[m
     }[m
[32m+[m
 }[m
[1mdiff --git opensearch/catalog-opensearch-source/src/main/resources/META-INF/spring/beans.xml opensearch/catalog-opensearch-source/src/main/resources/META-INF/spring/beans.xml[m
[1mindex b00d57c..64062aa 100644[m
[1m--- opensearch/catalog-opensearch-source/src/main/resources/META-INF/spring/beans.xml[m
[1m+++ opensearch/catalog-opensearch-source/src/main/resources/META-INF/spring/beans.xml[m
[36m@@ -31,9 +31,6 @@[m
 [m
     <osgi:reference id="filterAdapter" interface="ddf.catalog.filter.FilterAdapter"/>[m
 [m
[31m-    <osgi:reference id="securitySettings" interface="ddf.security.settings.SecuritySettingsService"/>[m
[31m-[m
[31m-[m
     <osgix:managed-service-factory id="DDFfederationPrototypeServiceFactory"[m
                                    factory-pid="OpenSearchSource"[m
                                    update-strategy="container-managed">[m
[36m@@ -54,7 +51,6 @@[m
             <beans:property name="localQueryOnly" value="false"/>[m
             <beans:property name="shouldConvertToBBox" value="true"/>[m
             <beans:property name="inputTransformer" ref="xmlInputTransformer"/>[m
[31m-            <beans:property name="securitySettings" ref="securitySettings"/>[m
             <beans:property name="parameters" value="q,src,mr,start,count,mt,dn,lat,lon,radius,bbox,polygon,dtstart,dtend,dateName,filter,sort" />[m
             <beans:constructor-arg ref="filterAdapter"/>[m
         </beans:bean>[m
[1mdiff --git opensearch/catalog-opensearch-source/src/test/java/ddf/catalog/source/opensearch/TestOpenSearchSource.java opensearch/catalog-opensearch-source/src/test/java/ddf/catalog/source/opensearch/TestOpenSearchSource.java[m
[1mindex 07e1fd4..0139de9 100644[m
[1m--- opensearch/catalog-opensearch-source/src/test/java/ddf/catalog/source/opensearch/TestOpenSearchSource.java[m
[1m+++ opensearch/catalog-opensearch-source/src/test/java/ddf/catalog/source/opensearch/TestOpenSearchSource.java[m
[36m@@ -1,61 +1,19 @@[m
 /**[m
  * Copyright (c) Codice Foundation[m
[31m- * [m
[32m+[m[32m *[m
  * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser[m
  * General Public License as published by the Free Software Foundation, either version 3 of the[m
  * License, or any later version.[m
[31m- * [m
[32m+[m[32m *[m
  * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without[m
  * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU[m
  * Lesser General Public License for more details. A copy of the GNU Lesser General Public License[m
  * is distributed along with this program and can be found at[m
  * <http://www.gnu.org/licenses/lgpl.html>.[m
[31m- * [m
[32m+[m[32m *[m
  **/[m
 package ddf.catalog.source.opensearch;[m
 [m
[31m-import static org.hamcrest.Matchers.containsString;[m
[31m-import static org.hamcrest.Matchers.is;[m
[31m-import static org.junit.Assert.assertThat;[m
[31m-import static org.junit.Assert.fail;[m
[31m-import static org.mockito.Matchers.any;[m
[31m-import static org.mockito.Matchers.eq;[m
[31m-import static org.mockito.Matchers.isA;[m
[31m-import static org.mockito.Mockito.mock;[m
[31m-import static org.mockito.Mockito.when;[m
[31m-[m
[31m-import java.io.ByteArrayInputStream;[m
[31m-import java.io.IOException;[m
[31m-import java.io.InputStream;[m
[31m-import java.io.Serializable;[m
[31m-import java.net.MalformedURLException;[m
[31m-import java.net.URI;[m
[31m-import java.net.URISyntaxException;[m
[31m-import java.net.URL;[m
[31m-import java.util.Arrays;[m
[31m-import java.util.HashMap;[m
[31m-import java.util.List;[m
[31m-import java.util.Map;[m
[31m-import java.util.Map.Entry;[m
[31m-import java.util.concurrent.ConcurrentHashMap;[m
[31m-[m
[31m-import javax.ws.rs.core.HttpHeaders;[m
[31m-import javax.ws.rs.core.MultivaluedHashMap;[m
[31m-import javax.ws.rs.core.MultivaluedMap;[m
[31m-import javax.ws.rs.core.Response;[m
[31m-[m
[31m-import junit.framework.Assert;[m
[31m-[m
[31m-import org.apache.commons.httpclient.NameValuePair;[m
[31m-import org.apache.commons.httpclient.util.ParameterParser;[m
[31m-import org.apache.cxf.jaxrs.client.Client;[m
[31m-import org.apache.cxf.jaxrs.client.WebClient;[m
[31m-import org.junit.Ignore;[m
[31m-import org.junit.Test;[m
[31m-import org.mockito.invocation.InvocationOnMock;[m
[31m-import org.mockito.stubbing.Answer;[m
[31m-import org.opengis.filter.Filter;[m
[31m-[m
 import ddf.catalog.data.BinaryContent;[m
 import ddf.catalog.data.Metacard;[m
 import ddf.catalog.data.Result;[m
[36m@@ -74,13 +32,52 @@[m
 import ddf.catalog.source.UnsupportedQueryException;[m
 import ddf.catalog.transform.CatalogTransformerException;[m
 import ddf.catalog.transform.InputTransformer;[m
[32m+[m[32mimport ddf.security.service.SecurityServiceException;[m
[32m+[m[32mimport junit.framework.Assert;[m
[32m+[m[32mimport org.apache.commons.httpclient.NameValuePair;[m
[32m+[m[32mimport org.apache.commons.httpclient.util.ParameterParser;[m
[32m+[m[32mimport org.apache.cxf.jaxrs.client.Client;[m
[32m+[m[32mimport org.apache.cxf.jaxrs.client.WebClient;[m
[32m+[m[32mimport org.junit.Ignore;[m
[32m+[m[32mimport org.junit.Test;[m
[32m+[m[32mimport org.mockito.invocation.InvocationOnMock;[m
[32m+[m[32mimport org.mockito.stubbing.Answer;[m
[32m+[m[32mimport org.opengis.filter.Filter;[m
[32m+[m
[32m+[m[32mimport javax.ws.rs.core.HttpHeaders;[m
[32m+[m[32mimport javax.ws.rs.core.MultivaluedHashMap;[m
[32m+[m[32mimport javax.ws.rs.core.MultivaluedMap;[m
[32m+[m[32mimport javax.ws.rs.core.Response;[m
[32m+[m[32mimport java.io.ByteArrayInputStream;[m
[32m+[m[32mimport java.io.IOException;[m
[32m+[m[32mimport java.io.InputStream;[m
[32m+[m[32mimport java.io.Serializable;[m
[32m+[m[32mimport java.net.MalformedURLException;[m
[32m+[m[32mimport java.net.URI;[m
[32m+[m[32mimport java.net.URISyntaxException;[m
[32m+[m[32mimport java.net.URL;[m
[32m+[m[32mimport java.util.Arrays;[m
[32m+[m[32mimport java.util.HashMap;[m
[32m+[m[32mimport java.util.List;[m
[32m+[m[32mimport java.util.Map;[m
[32m+[m[32mimport java.util.Map.Entry;[m
[32m+[m[32mimport java.util.concurrent.ConcurrentHashMap;[m
[32m+[m
[32m+[m[32mimport static org.hamcrest.Matchers.containsString;[m
[32m+[m[32mimport static org.hamcrest.Matchers.is;[m
[32m+[m[32mimport static org.junit.Assert.assertThat;[m
[32m+[m[32mimport static org.junit.Assert.fail;[m
[32m+[m[32mimport static org.mockito.Matchers.any;[m
[32m+[m[32mimport static org.mockito.Matchers.eq;[m
[32m+[m[32mimport static org.mockito.Matchers.isA;[m
[32m+[m[32mimport static org.mockito.Mockito.mock;[m
[32m+[m[32mimport static org.mockito.Mockito.when;[m
 [m
 /**[m
  * Tests parts of the {@link OpenSearchSource}[m
[31m- * [m
[32m+[m[32m *[m
  * @author Ashraf Barakat[m
  * @author ddf.isgs@lmco.com[m
[31m- * [m
  */[m
 public class TestOpenSearchSource {[m
 [m
[36m@@ -94,23 +91,128 @@[m
 [m
     private static FilterBuilder filterBuilder = new GeotoolsFilterBuilder();[m
 [m
[32m+[m[32m    private static InputStream getSampleAtomStream() {[m
[32m+[m[32m        String response =[m
[32m+[m[32m                "<feed xmlns=\"http://www.w3.org/2005/Atom\" xmlns:os=\"http://a9.com/-/spec/opensearch/1.1/\">\r\n"[m
[32m+[m[32m                        + "    <title type=\"text\">Query Response</title>\r\n"[m
[32m+[m[32m                        + "    <updated>2013-01-31T23:22:37.298Z</updated>\r\n"[m
[32m+[m[32m                        + "    <id>urn:uuid:a27352c9-f935-45f0-9b8c-5803095164bb</id>\r\n"[m
[32m+[m[32m                        + "    <link href=\"#\" rel=\"self\" />\r\n" + "    <author>\r\n"[m
[32m+[m[32m                        + "        <name>Codice</name>\r\n" + "    </author>\r\n"[m
[32m+[m[32m                        + "    <generator version=\"2.1.0.20130129-1341\">ddf123</generator>\r\n"[m
[32m+[m[32m                        + "    <os:totalResults>1</os:totalResults>\r\n"[m
[32m+[m[32m                        + "    <os:itemsPerPage>10</os:itemsPerPage>\r\n"[m
[32m+[m[32m                        + "    <os:startIndex>1</os:startIndex>\r\n"[m
[32m+[m[32m                        + "    <entry xmlns:relevance=\"http://a9.com/-/opensearch/extensions/relevance/1.0/\" xmlns:fs=\"http://a9.com/-/opensearch/extensions/federation/1.0/\"\r\n"[m
[32m+[m[32m                        + "        xmlns:georss=\"http://www.georss.org/georss\">\r\n"[m
[32m+[m[32m                        + "        <fs:resultSource fs:sourceId=\"ddf123\" />\r\n"[m
[32m+[m[32m                        + "        <relevance:score>0.19</relevance:score>\r\n"[m
[32m+[m[32m                        + "        <id>urn:catalog:id:ee7a161e01754b9db1872bfe39d1ea09</id>\r\n"[m
[32m+[m[32m                        + "        <title type=\"text\">F-15 lands in Libya; Crew Picked Up</title>\r\n"[m
[32m+[m[32m                        + "        <updated>2013-01-31T23:22:31.648Z</updated>\r\n"[m
[32m+[m[32m                        + "        <published>2013-01-31T23:22:31.648Z</published>\r\n"[m
[32m+[m[32m                        + "        <link href=\"http://123.45.67.123:8181/services/catalog/ddf123/ee7a161e01754b9db1872bfe39d1ea09\" rel=\"alternate\" title=\"View Complete Metacard\" />\r\n"[m
[32m+[m[32m                        + "        <category term=\"Resource\" />\r\n"[m
[32m+[m[32m                        + "        <georss:where xmlns:gml=\"http://www.opengis.net/gml\">\r\n"[m
[32m+[m[32m                        + "            <gml:Point>\r\n"[m
[32m+[m[32m                        + "                <gml:pos>32.8751900768792 13.1874561309814</gml:pos>\r\n"[m
[32m+[m[32m                        + "            </gml:Point>\r\n" + "        </georss:where>\r\n"[m
[32m+[m[32m                        + "        <content type=\"application/xml\">\r\n"[m
[32m+[m[32m                        + "            <ns3:metacard xmlns:ns3=\"urn:catalog:metacard\" xmlns:ns2=\"http://www.w3.org/1999/xlink\" xmlns:ns1=\"http://www.opengis.net/gml\"\r\n"[m
[32m+[m[32m                        + "                xmlns:ns4=\"http://www.w3.org/2001/SMIL20/\" xmlns:ns5=\"http://www.w3.org/2001/SMIL20/Language\" ns1:id=\"4535c53fc8bc4404a1d32a5ce7a29585\">\r\n"[m
[32m+[m[32m                        + "                <ns3:type>ddf.metacard</ns3:type>\r\n"[m
[32m+[m[32m                        + "                <ns3:source>ddf.distribution</ns3:source>\r\n"[m
[32m+[m[32m                        + "                <ns3:geometry name=\"location\">\r\n"[m
[32m+[m[32m                        + "                    <ns3:value>\r\n"[m
[32m+[m[32m                        + "                        <ns1:Point>\r\n"[m
[32m+[m[32m                        + "                            <ns1:pos>32.8751900768792 13.1874561309814</ns1:pos>\r\n"[m
[32m+[m[32m                        + "                        </ns1:Point>\r\n"[m
[32m+[m[32m                        + "                    </ns3:value>\r\n"[m
[32m+[m[32m                        + "                </ns3:geometry>\r\n"[m
[32m+[m[32m                        + "                <ns3:dateTime name=\"created\">\r\n"[m
[32m+[m[32m                        + "                    <ns3:value>2013-01-31T16:22:31.648-07:00</ns3:value>\r\n"[m
[32m+[m[32m                        + "                </ns3:dateTime>\r\n"[m
[32m+[m[32m                        + "                <ns3:dateTime name=\"modified\">\r\n"[m
[32m+[m[32m                        + "                    <ns3:value>2013-01-31T16:22:31.648-07:00</ns3:value>\r\n"[m
[32m+[m[32m                        + "                </ns3:dateTime>\r\n"[m
[32m+[m[32m                        + "                <ns3:stringxml name=\"metadata\">\r\n"[m
[32m+[m[32m                        + "                    <ns3:value>\r\n"[m
[32m+[m[32m                        + "                        <ns6:xml xmlns:ns6=\"urn:sample:namespace\" xmlns=\"urn:sample:namespace\">Example description.</ns6:xml>\r\n"[m
[32m+[m[32m                        + "                    </ns3:value>\r\n"[m
[32m+[m[32m                        + "                </ns3:stringxml>\r\n"[m
[32m+[m[32m                        + "                <ns3:string name=\"metadata-content-type-version\">\r\n"[m
[32m+[m[32m                        + "                    <ns3:value>myVersion</ns3:value>\r\n"[m
[32m+[m[32m                        + "                </ns3:string>\r\n"[m
[32m+[m[32m                        + "                <ns3:string name=\"metadata-content-type\">\r\n"[m
[32m+[m[32m                        + "                    <ns3:value>myType</ns3:value>\r\n"[m
[32m+[m[32m                        + "                </ns3:string>\r\n"[m
[32m+[m[32m                        + "                <ns3:string name=\"title\">\r\n"[m
[32m+[m[32m                        + "                    <ns3:value>Example title</ns3:value>\r\n"[m
[32m+[m[32m                        + "                </ns3:string>\r\n" + "            </ns3:metacard>\r\n"[m
[32m+[m[32m                        + "        </content>\r\n" + "    </entry>\r\n" + "</feed>";[m
[32m+[m[32m        return new ByteArrayInputStream(response.getBytes());[m
[32m+[m
[32m+[m[32m    }[m
[32m+[m
[32m+[m[32m    private static InputStream getBinaryData() {[m
[32m+[m
[32m+[m[32m        byte[] sampleBytes = {80, 81, 82};[m
[32m+[m
[32m+[m[32m        return new ByteArrayInputStream(sampleBytes);[m
[32m+[m[32m    }[m
[32m+[m
[32m+[m[32m    private static InputStream getSampleXmlStream() {[m
[32m+[m
[32m+[m[32m        String response = "\r\n"[m
[32m+[m[32m                + "            <ns3:metacard xmlns:ns3=\"urn:catalog:metacard\" xmlns:ns2=\"http://www.w3.org/1999/xlink\" xmlns:ns1=\"http://www.opengis.net/gml\"\r\n"[m
[32m+[m[32m                + "                xmlns:ns4=\"http://www.w3.org/2001/SMIL20/\" xmlns:ns5=\"http://www.w3.org/2001/SMIL20/Language\" ns1:id=\"4535c53fc8bc4404a1d32a5ce7a29585\">\r\n"[m
[32m+[m[32m                + "                <ns3:type>ddf.metacard</ns3:type>\r\n"[m
[32m+[m[32m                + "                <ns3:source>ddf.distribution</ns3:source>\r\n"[m
[32m+[m[32m                + "                <ns3:geometry name=\"location\">\r\n"[m
[32m+[m[32m                + "                    <ns3:value>\r\n" + "                        <ns1:Point>\r\n"[m
[32m+[m[32m                + "                            <ns1:pos>32.8751900768792 13.1874561309814</ns1:pos>\r\n"[m
[32m+[m[32m                + "                        </ns1:Point>\r\n"[m
[32m+[m[32m                + "                    </ns3:value>\r\n" + "                </ns3:geometry>\r\n"[m
[32m+[m[32m                + "                <ns3:dateTime name=\"created\">\r\n"[m
[32m+[m[32m                + "                    <ns3:value>2013-01-31T16:22:31.648-07:00</ns3:value>\r\n"[m
[32m+[m[32m                + "                </ns3:dateTime>\r\n"[m
[32m+[m[32m                + "                <ns3:dateTime name=\"modified\">\r\n"[m
[32m+[m[32m                + "                    <ns3:value>2013-01-31T16:22:31.648-07:00</ns3:value>\r\n"[m
[32m+[m[32m                + "                </ns3:dateTime>\r\n"[m
[32m+[m[32m                + "                <ns3:stringxml name=\"metadata\">\r\n"[m
[32m+[m[32m                + "                    <ns3:value>\r\n"[m
[32m+[m[32m                + "                        <ns6:xml xmlns:ns6=\"urn:sample:namespace\" xmlns=\"urn:sample:namespace\">Example description.</ns6:xml>\r\n"[m
[32m+[m[32m                + "                    </ns3:value>\r\n" + "                </ns3:stringxml>\r\n"[m
[32m+[m[32m                + "                <ns3:string name=\"metadata-content-type-version\">\r\n"[m
[32m+[m[32m                + "                    <ns3:value>myVersion</ns3:value>\r\n"[m
[32m+[m[32m                + "                </ns3:string>\r\n"[m
[32m+[m[32m                + "                <ns3:string name=\"metadata-content-type\">\r\n"[m
[32m+[m[32m                + "                    <ns3:value>myType</ns3:value>\r\n"[m
[32m+[m[32m                + "                </ns3:string>\r\n"[m
[32m+[m[32m                + "                <ns3:string name=\"title\">\r\n"[m
[32m+[m[32m                + "                    <ns3:value>Example title</ns3:value>\r\n"[m
[32m+[m[32m                + "                </ns3:string>\r\n" + "            </ns3:metacard>\r\n";[m
[32m+[m
[32m+[m[32m        return new ByteArrayInputStream(response.getBytes());[m
[32m+[m[32m    }[m
[32m+[m
     /**[m
      * Tests the proper query is sent to the remote source for query by id.[m
[31m-     * [m
[32m+[m[32m     *[m
      * @throws UnsupportedQueryException[m
      * @throws IOException[m
      * @throws MalformedURLException[m
      */[m
     @Test[m
[31m-    public void testQuery_ById() throws UnsupportedQueryException,[m
[31m-        IOException {[m
[32m+[m[32m    public void testQuery_ById()[m
[32m+[m[32m            throws UnsupportedQueryException, IOException, SecurityServiceException {[m
         WebClient client = mock(WebClient.class);[m
 [m
         Response clientResponse = mock(Response.class);[m
 [m
         OpenSearchConnection openSearchConnection = mock(OpenSearchConnection.class);[m
 [m
[31m-        when(openSearchConnection.getOpenSearchWebClient()).thenReturn(client);[m
[32m+[m[32m        when(openSearchConnection.getOpenSearchWebClient(null)).thenReturn(client);[m
 [m
         when(client.get()).thenReturn(clientResponse);[m
 [m
[36m@@ -118,18 +220,21 @@[m [mpublic void testQuery_ById() throws UnsupportedQueryException,[m
 [m
         Client proxy = mock(Client.class);[m
 [m
[31m-        when(openSearchConnection.newRestClient(any(String.class), any(Query.class), any(String.class), any(Boolean.class))).thenReturn(proxy);[m
[32m+[m[32m        when(openSearchConnection[m
[32m+[m[32m                .newRestClient(any(String.class), any(Query.class), any(String.class),[m
[32m+[m[32m                        any(Boolean.class))).thenReturn(proxy);[m
 [m
[31m-        when(openSearchConnection.getWebClientFromClient(proxy)).thenReturn(client);[m
[32m+[m[32m//        when(openSearchConnection.getWebClientFromClient(proxy)).thenReturn(client);[m
 [m
[31m-        when(clientResponse.getEntity()).thenReturn([m
[31m-                new BinaryContentImpl(getSampleXmlStream()).getInputStream());[m
[32m+[m[32m        when(clientResponse.getEntity())[m
[32m+[m[32m                .thenReturn(new BinaryContentImpl(getSampleXmlStream()).getInputStream());[m
 [m
         OpenSearchSource source = new OpenSearchSource(FILTER_ADAPTER);[m
         source.setInputTransformer(getMockInputTransformer());[m
         source.setEndpointUrl("http://localhost:8181/services/catalog/query");[m
         source.init();[m
[31m-        source.setParameters("q,src,mr,start,count,mt,dn,lat,lon,radius,bbox,polygon,dtstart,dtend,dateName,filter,sort");[m
[32m+[m[32m        source.setParameters([m
[32m+[m[32m                "q,src,mr,start,count,mt,dn,lat,lon,radius,bbox,polygon,dtstart,dtend,dateName,filter,sort");[m
 [m
         source.openSearchConnection = openSearchConnection;[m
 [m
[36m@@ -145,8 +250,9 @@[m [mpublic void testQuery_ById() throws UnsupportedQueryException,[m
     @Test[m
     @Ignore[m
     // Ignored because Content Type support has yet to be added.[m
[31m-    public void testQuery_ByContentType() throws UnsupportedQueryException,[m
[31m-        IOException, URISyntaxException {[m
[32m+[m[32m    public void testQuery_ByContentType()[m
[32m+[m[32m            throws UnsupportedQueryException, IOException, URISyntaxException,[m
[32m+[m[32m            SecurityServiceException {[m
 [m
         // given[m
         FirstArgumentCapture answer = new FirstArgumentCapture(getSampleAtomStream());[m
[36m@@ -166,28 +272,30 @@[m [mpublic void testQuery_ByContentType() throws UnsupportedQueryException,[m
     }[m
 [m
     @Test[m
[31m-    public void testQuery_BySearchPhrase() throws UnsupportedQueryException, URISyntaxException,[m
[31m-        IOException {[m
[32m+[m[32m    public void testQuery_BySearchPhrase()[m
[32m+[m[32m            throws UnsupportedQueryException, URISyntaxException, IOException,[m
[32m+[m[32m            SecurityServiceException {[m
         WebClient client = mock(WebClient.class);[m
 [m
         Response clientResponse = mock(Response.class);[m
 [m
         OpenSearchConnection openSearchConnection = mock(OpenSearchConnection.class);[m
 [m
[31m-        when(openSearchConnection.getOpenSearchWebClient()).thenReturn(client);[m
[32m+[m[32m        when(openSearchConnection.getOpenSearchWebClient(null)).thenReturn(client);[m
 [m
         when(client.get()).thenReturn(clientResponse);[m
 [m
         when(clientResponse.getStatus()).thenReturn(Response.Status.OK.getStatusCode());[m
[31m-        [m
[31m-        when(clientResponse.getEntity()).thenReturn([m
[31m-                new BinaryContentImpl(getSampleAtomStream()).getInputStream());[m
[32m+[m
[32m+[m[32m        when(clientResponse.getEntity())[m
[32m+[m[32m                .thenReturn(new BinaryContentImpl(getSampleAtomStream()).getInputStream());[m
 [m
         OpenSearchSource source = new OpenSearchSource(FILTER_ADAPTER);[m
         source.setInputTransformer(getMockInputTransformer());[m
         source.setEndpointUrl("http://localhost:8181/services/catalog/query");[m
         source.init();[m
[31m-        source.setParameters("q,src,mr,start,count,mt,dn,lat,lon,radius,bbox,polygon,dtstart,dtend,dateName,filter,sort");[m
[32m+[m[32m        source.setParameters([m
[32m+[m[32m                "q,src,mr,start,count,mt,dn,lat,lon,radius,bbox,polygon,dtstart,dtend,dateName,filter,sort");[m
 [m
         source.openSearchConnection = openSearchConnection;[m
 [m
[36m@@ -207,22 +315,23 @@[m [mpublic void testQuery_BySearchPhrase() throws UnsupportedQueryException, URISynt[m
     }[m
 [m
     @Test[m
[31m-    public void testQuery_BySearchPhrase_ContentTypeSet() throws UnsupportedQueryException, URISyntaxException,[m
[31m-        IOException {[m
[32m+[m[32m    public void testQuery_BySearchPhrase_ContentTypeSet()[m
[32m+[m[32m            throws UnsupportedQueryException, URISyntaxException, IOException,[m
[32m+[m[32m            SecurityServiceException {[m
         WebClient client = mock(WebClient.class);[m
 [m
         Response clientResponse = mock(Response.class);[m
 [m
         OpenSearchConnection openSearchConnection = mock(OpenSearchConnection.class);[m
 [m
[31m-        when(openSearchConnection.getOpenSearchWebClient()).thenReturn(client);[m
[32m+[m[32m        when(openSearchConnection.getOpenSearchWebClient(null)).thenReturn(client);[m
 [m
         when(client.get()).thenReturn(clientResponse);[m
 [m
         when(clientResponse.getStatus()).thenReturn(Response.Status.OK.getStatusCode());[m
[31m-        [m
[31m-        when(clientResponse.getEntity()).thenReturn([m
[31m-                new BinaryContentImpl(getSampleAtomStream()).getInputStream());[m
[32m+[m
[32m+[m[32m        when(clientResponse.getEntity())[m
[32m+[m[32m                .thenReturn(new BinaryContentImpl(getSampleAtomStream()).getInputStream());[m
 [m
         OpenSearchSource source = new OpenSearchSource(FILTER_ADAPTER);[m
         InputTransformer inputTransformer = mock(InputTransformer.class);[m
[36m@@ -234,8 +343,8 @@[m [mpublic void testQuery_BySearchPhrase_ContentTypeSet() throws UnsupportedQueryExc[m
 [m
         try {[m
             when(inputTransformer.transform(isA(InputStream.class))).thenReturn(generatedMetacard);[m
[31m-            when(inputTransformer.transform(isA(InputStream.class), isA(String.class))).thenReturn([m
[31m-                    generatedMetacard);[m
[32m+[m[32m            when(inputTransformer.transform(isA(InputStream.class), isA(String.class)))[m
[32m+[m[32m                    .thenReturn(generatedMetacard);[m
         } catch (IOException e) {[m
             fail();[m
         } catch (CatalogTransformerException e) {[m
[36m@@ -244,14 +353,15 @@[m [mpublic void testQuery_BySearchPhrase_ContentTypeSet() throws UnsupportedQueryExc[m
         source.setInputTransformer(inputTransformer);[m
         source.setEndpointUrl("http://localhost:8181/services/catalog/query");[m
         source.init();[m
[31m-        source.setParameters("q,src,mr,start,count,mt,dn,lat,lon,radius,bbox,polygon,dtstart,dtend,dateName,filter,sort");[m
[32m+[m[32m        source.setParameters([m
[32m+[m[32m                "q,src,mr,start,count,mt,dn,lat,lon,radius,bbox,polygon,dtstart,dtend,dateName,filter,sort");[m
 [m
         source.openSearchConnection = openSearchConnection;[m
 [m
         Filter filter = filterBuilder.attribute(Metacard.METADATA).like()[m
                 .text(SAMPLE_SEARCH_PHRASE);[m
         SourceResponse response = source.query(new QueryRequestImpl(new QueryImpl(filter)));[m
[31m-        [m
[32m+[m
         Assert.assertEquals(1, response.getHits());[m
         List<Result> results = response.getResults();[m
         Assert.assertTrue(results.size() == 1);[m
[36m@@ -262,8 +372,9 @@[m [mpublic void testQuery_BySearchPhrase_ContentTypeSet() throws UnsupportedQueryExc[m
     }[m
 [m
     @Test[m
[31m-    public void testQueryAnyText() throws UnsupportedQueryException, URISyntaxException,[m
[31m-        IOException {[m
[32m+[m[32m    public void testQueryAnyText()[m
[32m+[m[32m            throws UnsupportedQueryException, URISyntaxException, IOException,[m
[32m+[m[32m            SecurityServiceException {[m
         WebClient client = mock(WebClient.class);[m
 [m
         Response clientResponse = mock(Response.class);[m
[36m@@ -272,7 +383,7 @@[m [mpublic void testQueryAnyText() throws UnsupportedQueryException, URISyntaxExcept[m
 [m
         OpenSearchConnection openSearchConnection = mock(OpenSearchConnection.class);[m
 [m
[31m-        when(openSearchConnection.getOpenSearchWebClient()).thenReturn(client);[m
[32m+[m[32m        when(openSearchConnection.getOpenSearchWebClient(null)).thenReturn(client);[m
 [m
         when(client.get()).thenReturn(clientResponse);[m
 [m
[36m@@ -280,7 +391,8 @@[m [mpublic void testQueryAnyText() throws UnsupportedQueryException, URISyntaxExcept[m
         source.setInputTransformer(getMockInputTransformer());[m
         source.setEndpointUrl("http://localhost:8181/services/catalog/query");[m
         source.init();[m
[31m-        source.setParameters("q,src,mr,start,count,mt,dn,lat,lon,radius,bbox,polygon,dtstart,dtend,dateName,filter,sort");[m
[32m+[m[32m        source.setParameters([m
[32m+[m[32m                "q,src,mr,start,count,mt,dn,lat,lon,radius,bbox,polygon,dtstart,dtend,dateName,filter,sort");[m
 [m
         source.openSearchConnection = openSearchConnection;[m
 [m
[36m@@ -295,18 +407,18 @@[m [mpublic void testQueryAnyText() throws UnsupportedQueryException, URISyntaxExcept[m
     }[m
 [m
     @Test(expected = UnsupportedQueryException.class)[m
[31m-    public void testQueryBadResponse() throws UnsupportedQueryException,[m
[31m-            IOException {[m
[32m+[m[32m    public void testQueryBadResponse()[m
[32m+[m[32m            throws UnsupportedQueryException, IOException, SecurityServiceException {[m
         WebClient client = mock(WebClient.class);[m
 [m
         Response clientResponse = mock(Response.class);[m
 [m
[31m-        when(clientResponse.getStatus()).thenReturn([m
[31m-                Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());[m
[32m+[m[32m        when(clientResponse.getStatus())[m
[32m+[m[32m                .thenReturn(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());[m
 [m
         OpenSearchConnection openSearchConnection = mock(OpenSearchConnection.class);[m
 [m
[31m-        when(openSearchConnection.getOpenSearchWebClient()).thenReturn(client);[m
[32m+[m[32m        when(openSearchConnection.getOpenSearchWebClient(null)).thenReturn(client);[m
 [m
         when(client.get()).thenReturn(clientResponse);[m
 [m
[36m@@ -314,7 +426,8 @@[m [mpublic void testQueryBadResponse() throws UnsupportedQueryException,[m
         source.setInputTransformer(getMockInputTransformer());[m
         source.setEndpointUrl("http://localhost:8181/services/catalog/query");[m
         source.init();[m
[31m-        source.setParameters("q,src,mr,start,count,mt,dn,lat,lon,radius,bbox,polygon,dtstart,dtend,dateName,filter,sort");[m
[32m+[m[32m        source.setParameters([m
[32m+[m[32m                "q,src,mr,start,count,mt,dn,lat,lon,radius,bbox,polygon,dtstart,dtend,dateName,filter,sort");[m
 [m
         source.openSearchConnection = openSearchConnection;[m
 [m
[36m@@ -327,14 +440,15 @@[m [mpublic void testQueryBadResponse() throws UnsupportedQueryException,[m
 [m
     /**[m
      * Basic retrieve product case. Tests the url sent to the connection is correct.[m
[31m-     * [m
[32m+[m[32m     *[m
      * @throws ResourceNotSupportedException[m
      * @throws IOException[m
      * @throws ResourceNotFoundException[m
      */[m
     @Test[m
[31m-    public void testRetrieveResource() throws ResourceNotSupportedException, IOException,[m
[31m-        ResourceNotFoundException {[m
[32m+[m[32m    public void testRetrieveResource()[m
[32m+[m[32m            throws ResourceNotSupportedException, IOException, ResourceNotFoundException,[m
[32m+[m[32m            SecurityServiceException {[m
 [m
         // given[m
         FirstArgumentCapture answer = new FirstArgumentCapture(getBinaryData());[m
[36m@@ -351,10 +465,10 @@[m [mpublic void testRetrieveResource() throws ResourceNotSupportedException, IOExcep[m
         Assert.assertEquals(3, response.getResource().getByteArray().length);[m
     }[m
 [m
[31m-[m
     @Test[m
[31m-    public void testRetrieveResourceWithBytesToSkip() throws ResourceNotSupportedException, IOException,[m
[31m-            ResourceNotFoundException {[m
[32m+[m[32m    public void testRetrieveResourceWithBytesToSkip()[m
[32m+[m[32m            throws ResourceNotSupportedException, IOException, ResourceNotFoundException,[m
[32m+[m[32m            SecurityServiceException {[m
 [m
         // given[m
         FirstArgumentCapture answer = new FirstArgumentCapture(getBinaryData());[m
[36m@@ -376,18 +490,19 @@[m [mpublic void testRetrieveResourceWithBytesToSkip() throws ResourceNotSupportedExc[m
 [m
     /**[m
      * Given all null params, nothing will be returned, expect an exception.[m
[31m-     * [m
[32m+[m[32m     *[m
      * @throws ResourceNotSupportedException[m
      */[m
     @Test[m
[31m-    public void testRetrieveNullProduct() throws ResourceNotSupportedException, IOException {[m
[32m+[m[32m    public void testRetrieveNullProduct()[m
[32m+[m[32m            throws ResourceNotSupportedException, IOException, SecurityServiceException {[m
         WebClient client = mock(WebClient.class);[m
 [m
         Response clientResponse = mock(Response.class);[m
 [m
         OpenSearchConnection openSearchConnection = mock(OpenSearchConnection.class);[m
 [m
[31m-        when(openSearchConnection.getOpenSearchWebClient()).thenReturn(client);[m
[32m+[m[32m        when(openSearchConnection.getOpenSearchWebClient(null)).thenReturn(client);[m
 [m
         when(client.get()).thenReturn(clientResponse);[m
 [m
[36m@@ -414,14 +529,15 @@[m [mpublic void testRetrieveNullProduct() throws ResourceNotSupportedException, IOEx[m
     }[m
 [m
     @Test[m
[31m-    public void testRetrieveProductUriSyntaxException() throws ResourceNotSupportedException, IOException {[m
[32m+[m[32m    public void testRetrieveProductUriSyntaxException()[m
[32m+[m[32m            throws ResourceNotSupportedException, IOException, SecurityServiceException {[m
         WebClient client = mock(WebClient.class);[m
 [m
         Response clientResponse = mock(Response.class);[m
 [m
         OpenSearchConnection openSearchConnection = mock(OpenSearchConnection.class);[m
 [m
[31m-        when(openSearchConnection.getOpenSearchWebClient()).thenReturn(client);[m
[32m+[m[32m        when(openSearchConnection.getOpenSearchWebClient(null)).thenReturn(client);[m
 [m
         when(client.get()).thenReturn(clientResponse);[m
 [m
[36m@@ -446,20 +562,22 @@[m [mpublic void testRetrieveProductUriSyntaxException() throws ResourceNotSupportedE[m
             /*[m
              * this exception should have been thrown.[m
              */[m
[31m-            assertThat(e.getMessage(), containsString(OpenSearchSource.COULD_NOT_RETRIEVE_RESOURCE_MESSAGE));[m
[32m+[m[32m            assertThat(e.getMessage(),[m
[32m+[m[32m                    containsString(OpenSearchSource.COULD_NOT_RETRIEVE_RESOURCE_MESSAGE));[m
         }[m
 [m
     }[m
 [m
     @Test[m
[31m-    public void testRetrieveProductMalformedUrlException() throws ResourceNotSupportedException, IOException {[m
[32m+[m[32m    public void testRetrieveProductMalformedUrlException()[m
[32m+[m[32m            throws ResourceNotSupportedException, IOException, SecurityServiceException {[m
         WebClient client = mock(WebClient.class);[m
 [m
         Response clientResponse = mock(Response.class);[m
 [m
         OpenSearchConnection openSearchConnection = mock(OpenSearchConnection.class);[m
 [m
[31m-        when(openSearchConnection.getOpenSearchWebClient()).thenReturn(client);[m
[32m+[m[32m        when(openSearchConnection.getOpenSearchWebClient(null)).thenReturn(client);[m
 [m
         when(client.get()).thenReturn(clientResponse);[m
 [m
[36m@@ -484,7 +602,8 @@[m [mpublic void testRetrieveProductMalformedUrlException() throws ResourceNotSupport[m
             /*[m
              * this exception should have been thrown.[m
              */[m
[31m-            assertThat(e.getMessage(), containsString(OpenSearchSource.COULD_NOT_RETRIEVE_RESOURCE_MESSAGE));[m
[32m+[m[32m            assertThat(e.getMessage(),[m
[32m+[m[32m                    containsString(OpenSearchSource.COULD_NOT_RETRIEVE_RESOURCE_MESSAGE));[m
         }[m
 [m
     }[m
[36m@@ -501,25 +620,28 @@[m [mpublic void testQuery_QueryByMetacardIdFollowedByAnyTextQuery() throws Exception[m
 [m
         OpenSearchConnection openSearchConnection = mock(OpenSearchConnection.class);[m
 [m
[31m-        when(openSearchConnection.getOpenSearchWebClient()).thenReturn(client);[m
[32m+[m[32m        when(openSearchConnection.getOpenSearchWebClient(null)).thenReturn(client);[m
 [m
         when(client.get()).thenReturn(clientResponse);[m
 [m
         Client proxy = mock(Client.class);[m
 [m
[31m-        when(openSearchConnection.newRestClient(any(String.class), any(Query.class), any(String.class), any(Boolean.class))).thenReturn(proxy);[m
[32m+[m[32m        when(openSearchConnection[m
[32m+[m[32m                .newRestClient(any(String.class), any(Query.class), any(String.class),[m
[32m+[m[32m                        any(Boolean.class))).thenReturn(proxy);[m
 [m
[31m-        when(openSearchConnection.getWebClientFromClient(proxy)).thenReturn(client);[m
[32m+[m[32m//        when(openSearchConnection.getWebClientFromClient(proxy)).thenReturn(client);[m
 [m
[31m-        when(clientResponse.getEntity()).thenReturn([m
[31m-                new BinaryContentImpl(getSampleXmlStream()).getInputStream()).thenReturn([m
[31m-                new BinaryContentImpl(getSampleAtomStream()).getInputStream());[m
[32m+[m[32m        when(clientResponse.getEntity())[m
[32m+[m[32m                .thenReturn(new BinaryContentImpl(getSampleXmlStream()).getInputStream())[m
[32m+[m[32m                .thenReturn(new BinaryContentImpl(getSampleAtomStream()).getInputStream());[m
         OpenSearchSource source = new OpenSearchSource(FILTER_ADAPTER);[m
         source.setLocalQueryOnly(true);[m
         source.setInputTransformer(getMockInputTransformer());[m
         source.setEndpointUrl("http://localhost:8181/services/catalog/query");[m
         source.init();[m
[31m-        source.setParameters("q,src,mr,start,count,mt,dn,lat,lon,radius,bbox,polygon,dtstart,dtend,dateName,filter,sort");[m
[32m+[m[32m        source.setParameters([m
[32m+[m[32m                "q,src,mr,start,count,mt,dn,lat,lon,radius,bbox,polygon,dtstart,dtend,dateName,filter,sort");[m
 [m
         source.openSearchConnection = openSearchConnection;[m
 [m
[36m@@ -548,7 +670,7 @@[m [mprivate NameValuePair pair(String name, String value) {[m
     }[m
 [m
     private List<NameValuePair> extractQueryParams(FirstArgumentCapture answer)[m
[31m-        throws MalformedURLException, URISyntaxException {[m
[32m+[m[32m            throws MalformedURLException, URISyntaxException {[m
         URL url = new URI(answer.getInputArg()).toURL();[m
         ParameterParser paramParser = new ParameterParser();[m
         List<NameValuePair> pairs = paramParser.parse(url.getQuery(), '&');[m
[36m@@ -577,14 +699,14 @@[m [mprivate void verifyOpenSearchUrl(List<NameValuePair> pairs, NameValuePair... ans[m
 [m
     /**[m
      * Verifies that the rest of the entries don't have a corresponding value[m
[31m-     * [m
[32m+[m[32m     *[m
      * @param nvpMap[m
      */[m
     private void verifyAllEntriesBlank(ConcurrentHashMap<String, String> nvpMap) {[m
         for (Entry<String, String> entry : nvpMap.entrySet()) {[m
 [m
[31m-            String errorMessage = "[" + entry.getKey() + "]"[m
[31m-                    + " should not have a corresponding value.";[m
[32m+[m[32m            String errorMessage =[m
[32m+[m[32m                    "[" + entry.getKey() + "]" + " should not have a corresponding value.";[m
 [m
             assertThat(errorMessage, entry.getValue(), is(""));[m
         }[m
[36m@@ -600,26 +722,29 @@[m [mprivate void verifyAllEntriesBlank(ConcurrentHashMap<String, String> nvpMap) {[m
     }[m
 [m
     private OpenSearchSource givenSource(Answer<BinaryContent> answer)[m
[31m-        throws IOException {[m
[32m+[m[32m            throws IOException, SecurityServiceException {[m
         WebClient client = mock(WebClient.class);[m
 [m
         OpenSearchConnection openSearchConnection = mock(OpenSearchConnection.class);[m
 [m
         Client proxy = mock(Client.class);[m
 [m
[31m-        when(openSearchConnection.newRestClient(any(String.class), any(Query.class), any(String.class), any(Boolean.class))).thenReturn(proxy);[m
[32m+[m[32m        when(openSearchConnection[m
[32m+[m[32m                .newRestClient(any(String.class), any(Query.class), any(String.class),[m
[32m+[m[32m                        any(Boolean.class))).thenReturn(proxy);[m
 [m
[31m-        when(openSearchConnection.getWebClientFromClient(proxy)).thenReturn(client);[m
[32m+[m[32m//        when(openSearchConnection.getWebClientFromClient(proxy)).thenReturn(client);[m
 [m
         Response clientResponse = mock(Response.class);[m
 [m
         when(client.get()).thenReturn(clientResponse);[m
 [m
         when(clientResponse.getEntity()).thenReturn(getBinaryData());[m
[31m-        [m
[31m-        when(clientResponse.getHeaderString(eq(OpenSearchSource.HEADER_ACCEPT_RANGES))).thenReturn(OpenSearchSource.BYTES);[m
 [m
[31m-        when(openSearchConnection.getOpenSearchWebClient()).thenReturn(client);[m
[32m+[m[32m        when(clientResponse.getHeaderString(eq(OpenSearchSource.HEADER_ACCEPT_RANGES)))[m
[32m+[m[32m                .thenReturn(OpenSearchSource.BYTES);[m
[32m+[m
[32m+[m[32m        when(openSearchConnection.getOpenSearchWebClient(null)).thenReturn(client);[m
 [m
         MultivaluedMap<String, Object> headers = new MultivaluedHashMap<String, Object>();[m
         headers.put(HttpHeaders.CONTENT_TYPE, Arrays.<Object>asList("application/octet-stream"));[m
[36m@@ -628,7 +753,8 @@[m [mprivate OpenSearchSource givenSource(Answer<BinaryContent> answer)[m
 [m
         OpenSearchSource source = new OpenSearchSource(FILTER_ADAPTER);[m
         source.setEndpointUrl("http://localhost:8181/services/catalog/query");[m
[31m-        source.setParameters("q,src,mr,start,count,mt,dn,lat,lon,radius,bbox,polygon,dtstart,dtend,dateName,filter,sort");[m
[32m+[m[32m        source.setParameters([m
[32m+[m[32m                "q,src,mr,start,count,mt,dn,lat,lon,radius,bbox,polygon,dtstart,dtend,dateName,filter,sort");[m
         source.init();[m
         source.setLocalQueryOnly(true);[m
         source.setInputTransformer(getMockInputTransformer());[m
[36m@@ -643,8 +769,8 @@[m [mprotected InputTransformer getMockInputTransformer() {[m
 [m
         try {[m
             when(inputTransformer.transform(isA(InputStream.class))).thenReturn(generatedMetacard);[m
[31m-            when(inputTransformer.transform(isA(InputStream.class), isA(String.class))).thenReturn([m
[31m-                    generatedMetacard);[m
[32m+[m[32m            when(inputTransformer.transform(isA(InputStream.class), isA(String.class)))[m
[32m+[m[32m                    .thenReturn(generatedMetacard);[m
         } catch (IOException e) {[m
             fail();[m
         } catch (CatalogTransformerException e) {[m
[36m@@ -665,119 +791,11 @@[m [mprivate String getSample() {[m
         return "<xml></xml>";[m
     }[m
 [m
[31m-    private static InputStream getSampleAtomStream() {[m
[31m-        String response = "<feed xmlns=\"http://www.w3.org/2005/Atom\" xmlns:os=\"http://a9.com/-/spec/opensearch/1.1/\">\r\n"[m
[31m-                + "    <title type=\"text\">Query Response</title>\r\n"[m
[31m-                + "    <updated>2013-01-31T23:22:37.298Z</updated>\r\n"[m
[31m-                + "    <id>urn:uuid:a27352c9-f935-45f0-9b8c-5803095164bb</id>\r\n"[m
[31m-                + "    <link href=\"#\" rel=\"self\" />\r\n"[m
[31m-                + "    <author>\r\n"[m
[31m-                + "        <name>Codice</name>\r\n"[m
[31m-                + "    </author>\r\n"[m
[31m-                + "    <generator version=\"2.1.0.20130129-1341\">ddf123</generator>\r\n"[m
[31m-                + "    <os:totalResults>1</os:totalResults>\r\n"[m
[31m-                + "    <os:itemsPerPage>10</os:itemsPerPage>\r\n"[m
[31m-                + "    <os:startIndex>1</os:startIndex>\r\n"[m
[31m-                + "    <entry xmlns:relevance=\"http://a9.com/-/opensearch/extensions/relevance/1.0/\" xmlns:fs=\"http://a9.com/-/opensearch/extensions/federation/1.0/\"\r\n"[m
[31m-                + "        xmlns:georss=\"http://www.georss.org/georss\">\r\n"[m
[31m-                + "        <fs:resultSource fs:sourceId=\"ddf123\" />\r\n"[m
[31m-                + "        <relevance:score>0.19</relevance:score>\r\n"[m
[31m-                + "        <id>urn:catalog:id:ee7a161e01754b9db1872bfe39d1ea09</id>\r\n"[m
[31m-                + "        <title type=\"text\">F-15 lands in Libya; Crew Picked Up</title>\r\n"[m
[31m-                + "        <updated>2013-01-31T23:22:31.648Z</updated>\r\n"[m
[31m-                + "        <published>2013-01-31T23:22:31.648Z</published>\r\n"[m
[31m-                + "        <link href=\"http://123.45.67.123:8181/services/catalog/ddf123/ee7a161e01754b9db1872bfe39d1ea09\" rel=\"alternate\" title=\"View Complete Metacard\" />\r\n"[m
[31m-                + "        <category term=\"Resource\" />\r\n"[m
[31m-                + "        <georss:where xmlns:gml=\"http://www.opengis.net/gml\">\r\n"[m
[31m-                + "            <gml:Point>\r\n"[m
[31m-                + "                <gml:pos>32.8751900768792 13.1874561309814</gml:pos>\r\n"[m
[31m-                + "            </gml:Point>\r\n"[m
[31m-                + "        </georss:where>\r\n"[m
[31m-                + "        <content type=\"application/xml\">\r\n"[m
[31m-                + "            <ns3:metacard xmlns:ns3=\"urn:catalog:metacard\" xmlns:ns2=\"http://www.w3.org/1999/xlink\" xmlns:ns1=\"http://www.opengis.net/gml\"\r\n"[m
[31m-                + "                xmlns:ns4=\"http://www.w3.org/2001/SMIL20/\" xmlns:ns5=\"http://www.w3.org/2001/SMIL20/Language\" ns1:id=\"4535c53fc8bc4404a1d32a5ce7a29585\">\r\n"[m
[31m-                + "                <ns3:type>ddf.metacard</ns3:type>\r\n"[m
[31m-                + "                <ns3:source>ddf.distribution</ns3:source>\r\n"[m
[31m-                + "                <ns3:geometry name=\"location\">\r\n"[m
[31m-                + "                    <ns3:value>\r\n"[m
[31m-                + "                        <ns1:Point>\r\n"[m
[31m-                + "                            <ns1:pos>32.8751900768792 13.1874561309814</ns1:pos>\r\n"[m
[31m-                + "                        </ns1:Point>\r\n"[m
[31m-                + "                    </ns3:value>\r\n"[m
[31m-                + "                </ns3:geometry>\r\n"[m
[31m-                + "                <ns3:dateTime name=\"created\">\r\n"[m
[31m-                + "                    <ns3:value>2013-01-31T16:22:31.648-07:00</ns3:value>\r\n"[m
[31m-                + "                </ns3:dateTime>\r\n"[m
[31m-                + "                <ns3:dateTime name=\"modified\">\r\n"[m
[31m-                + "                    <ns3:value>2013-01-31T16:22:31.648-07:00</ns3:value>\r\n"[m
[31m-                + "                </ns3:dateTime>\r\n"[m
[31m-                + "                <ns3:stringxml name=\"metadata\">\r\n"[m
[31m-                + "                    <ns3:value>\r\n"[m
[31m-                + "                        <ns6:xml xmlns:ns6=\"urn:sample:namespace\" xmlns=\"urn:sample:namespace\">Example description.</ns6:xml>\r\n"[m
[31m-                + "                    </ns3:value>\r\n"[m
[31m-                + "                </ns3:stringxml>\r\n"[m
[31m-                + "                <ns3:string name=\"metadata-content-type-version\">\r\n"[m
[31m-                + "                    <ns3:value>myVersion</ns3:value>\r\n"[m
[31m-                + "                </ns3:string>\r\n"[m
[31m-                + "                <ns3:string name=\"metadata-content-type\">\r\n"[m
[31m-                + "                    <ns3:value>myType</ns3:value>\r\n"[m
[31m-                + "                </ns3:string>\r\n"[m
[31m-                + "                <ns3:string name=\"title\">\r\n"[m
[31m-                + "                    <ns3:value>Example title</ns3:value>\r\n"[m
[31m-                + "                </ns3:string>\r\n"[m
[31m-                + "            </ns3:metacard>\r\n"[m
[31m-                + "        </content>\r\n" + "    </entry>\r\n" + "</feed>";[m
[31m-        return new ByteArrayInputStream(response.getBytes());[m
[31m-[m
[31m-    }[m
[31m-[m
[31m-    private static InputStream getBinaryData() {[m
[31m-[m
[31m-        byte[] sampleBytes = {80, 81, 82};[m
[31m-[m
[31m-        return new ByteArrayInputStream(sampleBytes);[m
[31m-    }[m
[31m-[m
[31m-    private static InputStream getSampleXmlStream() {[m
[31m-[m
[31m-        String response = "\r\n"[m
[31m-                + "            <ns3:metacard xmlns:ns3=\"urn:catalog:metacard\" xmlns:ns2=\"http://www.w3.org/1999/xlink\" xmlns:ns1=\"http://www.opengis.net/gml\"\r\n"[m
[31m-                + "                xmlns:ns4=\"http://www.w3.org/2001/SMIL20/\" xmlns:ns5=\"http://www.w3.org/2001/SMIL20/Language\" ns1:id=\"4535c53fc8bc4404a1d32a5ce7a29585\">\r\n"[m
[31m-                + "                <ns3:type>ddf.metacard</ns3:type>\r\n"[m
[31m-                + "                <ns3:source>ddf.distribution</ns3:source>\r\n"[m
[31m-                + "                <ns3:geometry name=\"location\">\r\n"[m
[31m-                + "                    <ns3:value>\r\n"[m
[31m-                + "                        <ns1:Point>\r\n"[m
[31m-                + "                            <ns1:pos>32.8751900768792 13.1874561309814</ns1:pos>\r\n"[m
[31m-                + "                        </ns1:Point>\r\n"[m
[31m-                + "                    </ns3:value>\r\n"[m
[31m-                + "                </ns3:geometry>\r\n"[m
[31m-                + "                <ns3:dateTime name=\"created\">\r\n"[m
[31m-                + "                    <ns3:value>2013-01-31T16:22:31.648-07:00</ns3:value>\r\n"[m
[31m-                + "                </ns3:dateTime>\r\n"[m
[31m-                + "                <ns3:dateTime name=\"modified\">\r\n"[m
[31m-                + "                    <ns3:value>2013-01-31T16:22:31.648-07:00</ns3:value>\r\n"[m
[31m-                + "                </ns3:dateTime>\r\n"[m
[31m-                + "                <ns3:stringxml name=\"metadata\">\r\n"[m
[31m-                + "                    <ns3:value>\r\n"[m
[31m-                + "                        <ns6:xml xmlns:ns6=\"urn:sample:namespace\" xmlns=\"urn:sample:namespace\">Example description.</ns6:xml>\r\n"[m
[31m-                + "                    </ns3:value>\r\n"[m
[31m-                + "                </ns3:stringxml>\r\n"[m
[31m-                + "                <ns3:string name=\"metadata-content-type-version\">\r\n"[m
[31m-                + "                    <ns3:value>myVersion</ns3:value>\r\n"[m
[31m-                + "                </ns3:string>\r\n"[m
[31m-                + "                <ns3:string name=\"metadata-content-type\">\r\n"[m
[31m-                + "                    <ns3:value>myType</ns3:value>\r\n"[m
[31m-                + "                </ns3:string>\r\n"[m
[31m-                + "                <ns3:string name=\"title\">\r\n"[m
[31m-                + "                    <ns3:value>Example title</ns3:value>\r\n"[m
[31m-                + "                </ns3:string>\r\n"[m
[31m-                + "            </ns3:metacard>\r\n";[m
[32m+[m[32m    private class FirstArgumentCapture implements Answer<BinaryContent> {[m
 [m
[31m-        return new ByteArrayInputStream(response.getBytes());[m
[31m-    }[m
[32m+[m[32m        private InputStream returnInputStream;[m
 [m
[31m-    private class FirstArgumentCapture implements Answer<BinaryContent> {[m
[32m+[m[32m        private String inputArg;[m
 [m
         public FirstArgumentCapture() {[m
             this.returnInputStream = getSampleXmlStream();[m
[36m@@ -787,10 +805,6 @@[m [mpublic FirstArgumentCapture(InputStream inputStream) {[m
             this.returnInputStream = inputStream;[m
         }[m
 [m
[31m-        private InputStream returnInputStream;[m
[31m-[m
[31m-        private String inputArg;[m
[31m-[m
         public String getInputArg() {[m
             return inputArg;[m
         }[m